---
title: "1vs1_ROSI_DMSO"
author: "Taylor"
date: "1/21/2022"
output: html_document
---

#1vs1 ROSI DMSO
This is the RMArkdown for my ROSI vs DMSO data. I will be subsetting these by day and by 
```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = "~/Documents/Git/MSC_Bulk/RNA_seq_Data/")
suppressPackageStartupMessages({
  library(pasilla)
  library(pheatmap)
  library(RColorBrewer)
  library(DESeq2)
  library(BSgenome.Hsapiens.UCSC.hg38)
  library(tidyverse)
  library(dbplyr)
  library(annotables)
  library(org.Hs.eg.db)
  library(AnnotationDbi)
  library(TxDb.Hsapiens.UCSC.hg38.knownGene)
  library(EnsDb.Hsapiens.v86)
})
```
```{r load variables}
load("./DESeq2_Variables_Preanalyzed.RData") #this will have all of the preanalyzed data
```
##loading in data
```{r loading basic analysis variables}
#load(file = "~/Documents/Git/MSC_Bulk/RNA_seq_Data/ROSI_data.Rdata")This is the old data
load(file = "~/Documents/Git/MSC_Bulk/RNA_seq_Data/ROSI_noMSC_data.Rdata")
#This variable has all of the variables I created so hte RMD doesn't have to rerun them 
```

#Basic analysis
This starts out with the basic analysis of all the sample for with the condition(ROSI,ROSI,control) as the object that the program is comparing the samples with. 
I will remove all of the outlier samples that arn't clustering well with the other days. 
The RMArkdowns that I need to make after this are:
  *1vs1 comparisons for ROSI vs DMSO by day
  *1vs1 comparisons for ROSI vs DMSO by day
These will help to show that the most variability between the Condition and control. So I will subset the data. 


```{r, eval=FALSE}
#BiocManager::install("DESeq2")
suppressPackageStartupMessages(library("DESeq2"))

#Subsetting the data to only have the pertinent information
cts_mat_ROSI<- cts_mat[,which(substr(colnames(cts_mat),2,2)=="R"|substr(colnames(cts_mat),2,2)=="D")]
rownames(cts_mat_ROSI) <- substr(rownames(cts_mat_ROSI),1,15)#Removing the .# on the gene names

coldata_ROSI <- coldata_norm[which(substr(rownames(coldata_norm),2,2)=="R"|substr(rownames(coldata_norm),2,2)=="D"),]
dim(cts_mat_ROSI) #58037 27
dim(coldata_ROSI) #27 5

##Creating DDS object from the subset data                                  

dds_ROSI <- DESeqDataSetFromMatrix(countData = cts_mat_ROSI,
                                                 colData = coldata_ROSI,
                                                 design = ~ Time + condition                                               )
summary(dds_ROSI)
head(dds_ROSI)   

```

Add in feature data using metadata columns of DESeq object

```{r Adding feature/metadata to a dds object,eval=FALSE}
#The data I have nas a laging period and number after the ensmble gene name so I am removind that first then rerunning it all 
rownames(dds_ROSI) <- substr(rownames(dds_ROSI),1,15)
featureData <- data.frame(gene=substr(rownames(cts_mat_ROSI),1,15))
mcols(dds_ROSI) <- DataFrame(mcols(dds_ROSI), featureData)
mcols(dds_ROSI)
dim(featureData)
featureData$symbol <- cts_1021$SYMBOL# Adding mcol symbol data
mcols(dds_ROSI) <- DataFrame(mcols(dds_ROSI), featureData)
mcols(dds_ROSI)
```

Pre- filtering. Pre-filtering of count data, while not necessary for
analysis at this point, can help to minimize processing time of DESeq
functions!

## Filtering DDS Object

```{r Filtering a dds object,eval=FALSE}
keep <- rowSums(counts(dds_ROSI)) >= 50 #24717 genes passed this low bar
dds_ROSI <- dds_ROSI[keep,]
```

Setting factor levels. It can be useful to designate factor levels while
formatting your dds object since it will influence the manner in which
DESeq compares different groups within your data.

You can specify levels either using factor()

```{r Generating dds object, eval=FALSE}
dds_ROSI$condition <- factor(dds_ROSI$condition, levels = c("control","ROSI"))
dds_ROSI$condition
```

Bonus: In order to map the Ensembl IDs assigned to the genes back to
their gene symbol install the org.Dm.eg.eb library and the AnnotationDbi
library and use the function mapIds.


\#NORMALIZE: This is self-explanitory. This will normalize the data so
that we aren't getting a lot of outliers.

```{r normalize }
# Determine the size factors to use for normalization
dds_ROSI<- estimateSizeFactors(dds_ROSI)
sizeFactors(dds_ROSI)
# Extract the normalized counts
normalized_counts_ROSI <- counts(dds_ROSI, normalized=T)
head(normalized_counts_ROSI)
```

##Clustering Variace
VST stands for Variance stablizind transformation which is based on negative binomial data with a dispetion mean trend. 
Genes with high counts will haev similar VST and Rlog results to the ordinary log2 tranformaiton of the normalized counts. Genes with lower counts will be flattened out this allows us to use the VST and rlog data for better PCAs. 
###VST vs rlog
VST is much faster ot compute and less sensitive to outliers than the rlog. rlog is better for smaller datasets with samples sizeds below ~30. you can perform both VST and rlog if you are unsure and then compare the meanSdPlot  or PCAs to see which generates better data. 
```{r correlation heatmaps}
suppressPackageStartupMessages(library(pheatmap))

# Transform the normalized counts 
vsd_ROSI <- vst(dds_ROSI,blind=F )

# Extract the matrix of transformed counts
vsd_mat_ROSI <- assay(vsd_ROSI)

# Compute the correlation values between samples
vsd_cor_ROSI <- cor(vsd_mat_ROSI) 

# Plot the heatmap
pheatmap(vsd_cor_ROSI)
```

\#PCA time

```{r pca of all RNA-seq data}
# Transform the normalized counts 
#we did this last time so we can just use it again
#vsd_ROSI <- vst(dds_ROSI,blind=T)

#determine what to dolor the data by 
colData(vsd_ROSI)
# Plot the PCA of PC1 and PC2
plotPCA(vsd_ROSI, intgroup= "condition")
```

\#Looking at the other PCs

```{r}
plotPCA(vsd_ROSI, intgroup = c("Time","condition"))
```

\#Checing Dispersion

```{r Dispersion, eval=FALSE}
# Plot dispersions
# This is not working for me and I am not sure why 
plotDispEsts(dds_ROSI)
```
```{r Differential Expression calc, echo=FALSE, eval= FALSE}
#We didn't need to remave any of the samples so the basic analyis will be the same by conditon
# dds_ROSI <- DESeqDataSetFromMatrix(countData = cts_mat,
#                                                  colData = coldata_norm,
#                                                  design = ~ condition
#                                                  )
DES_ROSI <- DESeq(dds_ROSI)
res_ROSI <- results(DES_ROSI)
head(res_ROSI)
mcols(res)
```
## Filtering results

I am adding a specific contrast and alpha values to help remove some of the extra low hanging fruit.  
```{r Differential Expression results}
res_contr_ROSI <- results(DES_ROSI, contrast=c("condition","ROSI","control"),alpha = 0.05)
head(res_contr_ROSI)
```

Log fold change shrinkage for visualization and ranking This will help to show which genes are important and which aren't.

```{r Differential Expression logfold, eval=FALSE}
#BiocManager::install("apeglm")
library(apeglm)
# Shrink the log2 fold change estimates to be more accurate
res_contr_ROSI_lfc <- results(DES_ROSI, 
                        contrast=c("condition","ROSI","control"),
                        alpha = 0.05,
                        lfcThreshold = 0.2)
summary(res_contr_ROSI_lfc)

resLFC <- lfcShrink(dds_ROSI,
                    contrast = c("condition","ROSI","control"),
                    res = res_cont_ROSI_lfc)
summary(res_cont_ROSI_lfc)
#Save results as a dataframe
res_cont_ROSI_alldays_lfc.2 <- data.frame(res_cont_ROSI_lfc)
```


p-values and adjusted p-values

Reorganize your results by p-value, place the smaller p-values at the
top of your results list.

```{r Order results by padjusted values}
#res_Ordered <- res_cont_ROSI[order(res_cont_ROSI$pvalue),]
#summary(res_Ordered)
```

This is a lot of differentially expressed genes I will run through these
really quickly and then rerun them with a more stringent alpha and an
log2fold change cutoff added.

```{r, echo=FALSE}
sum(res_contr_ROSI$padj < 0.05, na.rm=TRUE) #114 lowering the Padjusted value removed some of the genes
sum(res_contr_ROSI$padj < 0.1, na.rm=TRUE) #210
res_contr_ROSI_padj_0.05 <- res_contr_ROSI[which(res_contr_ROSI$padj<0.05),]
res_contr_ROSI_padj_0.1 <- res_contr_ROSI[which(res_contr_ROSI$padj<0.1),]
sum(is.na(res_contr_ROSI_padj_0.1$padj))
```

##Annotate Results

```{r Annotate, eval=FALSE}
#install.packages("devtools")
#devtools::install_github("stephenturner/annotables")
library(dbplyr)
library(annotables)
grch38 # This is the genome build that I will be using for the annotation

res_cont_ROSI_all <- data.frame(res_contr_ROSI) %>% rownames_to_column(var = "genes")
res_cont_ROSI_all$ensgene <- substr(res_cont_ROSI_all$genes,1,15)
 
res_cont_ROSI_all <- left_join(x=res_cont_ROSI_all, y = grch38[,c("ensgene","symbol","description")],  by="ensgene")

head(res_cont_ROSI_all)
ROSI_only_sig <- res_cont_ROSI_all[res_cont_ROSI_all$padj<0.5,]
grep(pattern="PPAR",ROSI_only_sig$symbol)#1171
ROSI_only_sig$symbol[1171] #PPARD
grep(pattern="PPAR",res_cont_ROSI_all$symbol)
res_cont_ROSI_all[c(3158,3407,5490,8251,12756),]

```
```{r look at leptin}
head(res_cont_ROSI_all)
lepsrescont <- grep(res_cont_ROSI_all$symbol, pattern = "LEP")
res_cont_ROSI_alltime[lepsrescont,]

```


This brought down the number of genes from Exploring and exporting
results

Generate an MA plot of your results using the plotMA function. What are
the main components of this plot. What do the triangular points
represent?

```{r, echo=FALSE}
plotMA(res_contr_ROSI_padj_0.1, ylim=c(-4,4))
head(order(res_contr_ROSI_padj_0.1,padj))
head(res_contr_ROSI_padj_0.1)

```

Example of most significantly altered genes
```{r, echo=FALSE}
plotCounts(DES_ROSI, gene=which.min(res_contr_ROSI_padj_0.1$padj), intgroup="condition")
```

```{r Data transformations and visualization}
vsd_ROSI_ordered <- vst(DES_ROSI, blind=FALSE)
rld_ROSI <- rlog(DES_ROSI, blind=FALSE)
head(assay(vsd_ROSI_ordered), 5)
```
```{r variance}
#BiocManager::install("vsn")
# this gives log2(n + 1)
ntd_ROSI <- normTransform(DES_ROSI)
suppressPackageStartupMessages(library("vsn"))
meanSdPlot(assay(ntd_ROSI))
```

```{r Heatmap of the count matrix}
suppressPackageStartupMessages(library("pheatmap"))
select <- order(rowMeans(counts(DES_ROSI,normalized=TRUE)),
                decreasing=TRUE)[1:20]

df <- as.data.frame(colData(DES_ROSI)[,c("condition","Time")])
pheatmap(assay(ntd_ROSI)[select,], cluster_rows=TRUE, show_rownames=FALSE,
         cluster_cols=TRUE, annotation_col=df)
```
This is to see how similar the samples are
#Visualization Results 
The above work is all qualty control for the comparison of all of hte data. I will start looking at the differentially called genes for this comparison with  creating heatmaps of differentially expressed genes and then a chart of how they are being altered. 
##Heatmap  

```{r Significant normalized counts,eval= F}
#All significant normalized counts table
res_order_ROSI <-res_contr_ROSI[order(res_contr_ROSI$padj),]
res_sig_order_ROSI_padj_0.1 <-res_contr_ROSI_padj_0.1[order(res_contr_ROSI_padj_0.1$padj),]
res_sig_order_ROSI_padj_0.05 <- res_contr_ROSI_padj_0.05[order(res_contr_ROSI_padj_0.05$padj),]
dim(res_order_ROSI)#210 6
res_sig_normalized_counts_ROSI <- vsd_mat_ROSI[substr(rownames(res_sig_order_ROSI_padj_0.05),1,15),] 
dim(res_sig_normalized_counts_ROSI) #114 27
##Significant in each wat tables
res_ROSI_sig_down <-res_contr_ROSI_padj_0.1[which(res_contr_ROSI_padj_0.1$log2FoldChange<0),] 
dim(res_ROSI_sig_down) #145 6
res_ROSI_sig_up <-res_contr_ROSI_padj_0.1[which(res_contr_ROSI_padj_0.1$log2FoldChange>0),]
dim(res_ROSI_sig_up) #58 6

#Normalized counts table for up and down regulations
res_sig_normalized_counts_ROSI_up <- vsd_mat_ROSI[rownames(res_ROSI_sig_up[order(res_ROSI_sig_up$padj),]),]
res_sig_normalized_counts_ROSI_Down <- vsd_mat_ROSI[rownames(res_ROSI_sig_down[order(res_ROSI_sig_down$padj),]),] 

##Annnotate significant table
suppressPackageStartupMessages(library(org.Hs.eg.db))
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(EnsDb.Hsapiens.v86)
edb <- EnsDb.Hsapiens.v86
organism(edb)
keytypes(edb)
keys(edb,keytype = "GENEID")
###Annotate results table#####
res_ROSI_noMSC_anno <- data.frame(res_order_ROSI)
res_ROSI_noMSC_anno$symbol <- mapIds(edb,keys=rownames(res_order_ROSI),column="SYMBOL",keytype="GENEID",multiVals="first")
####DEGs####
ROSI_padj0.05_DEGS_anno <- data.frame(res_contr_ROSI_padj_0.05)
ROSI_padj0.05_DEGS_anno$symbol<-                                                                                      mapIds(edb,keys=rownames(res_contr_ROSI_padj_0.05),column="SYMBOL",keytype="GENEID",multiVals="first")
ROSI_padj0.05_DEGS_anno$symbol
ROSI_padj0.05_DEGS_anno <- ROSI_padj0.05_DEGS_anno[which(rownames(ROSI_padj0.05_DEGS_anno) %in% rownames(res_sig_order_ROSI_padj_0.05)),]
dim(ROSI_padj0.05_DEGS_anno)##118
upreg_ROSI_DEGS_anno_padj0.05 <- ROSI_padj0.05_DEGS_anno[which(ROSI_padj0.05_DEGS_anno$log2FoldChange>0),]
### Write out the full significant table as well as the full normalized counts table
library(writexl)
writexl::write_xlsx(ROSI_DEGS_anno,"~/Documents/Git/MSC_Bulk/RNA_seq_Data/ROSI_noMSC_DEGs_all_anno.xlsx")
```
```{r Saving Variables up to this point, eval= FALSE}
save(list = c("res_sig_order_ROSI","res_sig_normalized_counts_ROSI","res_ROSI_sig_down","res_ROSI_sig_up","normalized_counts_ROSI","DES_ROSI","res_contr_ROSI","res_contr_ROSI","vsd_ROSI","res_ROSI","dds_ROSI","cts_mat_ROSI","coldata_ROSI","res_sig_normalized_counts_ROSI_up","res_sig_normalized_counts_ROSI_Down","ROSI_DEGS_anno_padj0.05","res_ROSI_noMSC_anno","res_sig_order_ROSI_padj_0.05"),file = "~/Documents/Git/MSC_Bulk/RNA_seq_Data/ROSI_noMSC_data.Rdata")
```
###Heatmaps of noMSC DEGs subset by days
```{r Normalized significant genes Heatmaps}
#head(res_SD_D0_D14_sig)
# Choose a color palette from RColorBrewer
library(RColorBrewer)
heat_colors <- brewer.pal(6,"YlOrRd")
#display.brewer.all()

# Run pheatmap
pheatmap(res_sig_normalized_counts_ROSI,
         color = heat_colors,
         cluster_rows =T,
         cluster_cols = T,
         show_rownames =F,
         annotation = select(coldata_ROSI, Day, condition),
         scale ="row")
#colnames(res_sig_normalized_counts_ROSI)
###DAy 3 and day 14
pheatmap(res_sig_normalized_counts_ROSI[,c(1:6,22:27)],
         cluster_rows =T,
         color = heat_colors,
         cluster_cols = F,
         show_rownames =F,
         annotation = select(coldata_ROSI, condition, Day),
         scale = "row")
###Day 3 only
pheatmap(res_sig_normalized_counts_ROSI[,1:6],
         cluster_rows =T,
         cluster_cols = F,
         show_rownames =F,
         annotation = select(coldata_ROSI, condition, Day),
         scale = "row")
colnames(res_sig_normalized_counts_ROSI)
###Day 14 only
pheatmap(res_sig_normalized_counts_ROSI[,22:27],
         cluster_rows =T,
         color = heat_colors,
         cluster_cols = F,
         show_rownames =F,
         annotation = select(coldata_ROSI, condition, Day),
         scale = "row")
colnames(res_sig_normalized_counts_ROSI)

```

###Up and down regulated genes
##Top variance 
```{r top Variance genes}
library("genefilter")
#Lets us thee variance tables to get the samples that had the biggest discrepencey for day 0 to day 14. Then crease a heatmap to visualize the top genes that plat this role. We can use this to see which of the MSC s playted a larger role in the definithion of MSCs by what genes 
VarGenes_ROSI <- head(order(rowVars(assay(vsd_ROSI)),decreasing = TRUE),50)

#Create a matrix of the top genes 
VG_mat_ROSI  <- assay(vsd_ROSI)[VarGenes_SD_NE,]
head(vsd_ROSI)
VG_mat_ROSI  <- VG_mat_ROSI - rowMeans(VG_mat_ROSI)
anno_ROSI_var <- as.data.frame(colData(vsd_ROSI)[, c("condition","Time")])

library(AnnotationDbi)
library(org.Hs.eg.db)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(EnsDb.Hsapiens.v86)
edb <- EnsDb.Hsapiens.v86


## Annotating significant table
dim(VG_mat_ROSI)#normalizaerd data ROSI no MSCs

VG_mat_ROSI_annoDbi <- data.frame(VG_mat_ROSI)
VG_mat_ROSI_annoDbi$symbol<-                                                                                      mapIds(edb,keys=rownames(VG_mat_ROSI_annoDbi),column="SYMBOL",keytype="GENEID",multiVals="first")
dim(VG_mat_ROSI_annoDbi) #13013 7 
rownames(VG_mat_ROSI) <- VG_mat_SD_NE_annoDbi$symbol
pheatmap(VG_mat_ROSI, annotation_col = anno_ROSI_var)
```

###Top genes
```{r top genes}
##Creating Counts table with top DEGs##
top_ROSI <- data.frame(res_sig_normalized_counts_ROSI[1:20,]) %>% rownames_to_column(var ="ensgene")
dim(top_ROSI)
head(top_ROSI)
top_ROSI <- gather(top_ROSI, key ="samplename", value ="normalized_counts",2:28)
head(top_ROSI)

top_ROSI<- inner_join(top_ROSI,
                     rownames_to_column(coldata_ROSI, var ="samplename"),
                     by ="samplename")
head(top_ROSI)
top_ROSI_anno <- data.frame(top_ROSI)
top_ROSI_anno$symbol<-                                                                                      mapIds(edb,keys=top_ROSI$ensgene,column="SYMBOL",keytype="GENEID",multiVals="first")
top_ROSI_anno$symbol

ggplot(top_ROSI_anno)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top DE Genes ROSI vs DMSO")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))
## Top Geens with most variance
VG_mat_ROSI_annoDbi
top_var_ROSI <- data.frame(VG_mat_ROSI_annoDbi) %>% rownames_to_column(var ="ensgene")
dim(top_var_ROSI)
head(top_var_ROSI)
top_var_ROSI <- gather(top_var_ROSI, key ="samplename", value ="normalized_counts",2:28)
head(top_var_ROSI)

top_var_ROSI<- inner_join(top_var_ROSI,
                     rownames_to_column(coldata_ROSI, var ="samplename"),
                     by ="samplename")
head(top_var_ROSI)
top_var_ROSI_anno <- data.frame(top_var_ROSI)
top_var_ROSI_anno$symbol<-                                                                                      mapIds(edb,keys=top_var_ROSI$ensgene,column="SYMBOL",keytype="GENEID",multiVals="first")
top_var_ROSI_anno$symbol

ggplot(top_var_ROSI_anno)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top DE Genes ROSI vs DMSO")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))

####Up Regulated DEGS####
dim(res_sig_normalized_counts_ROSI_up) #54 27
top_ROSI_Up <- data.frame(res_sig_normalized_counts_ROSI_up) %>% rownames_to_column(var ="ensgene")
dim(top_ROSI_Up)
head(top_ROSI_Up)
top_ROSI_Up <- gather(top_ROSI_Up, key ="samplename", value ="normalized_counts",2:28)
head(top_ROSI_Up)

top_ROSI_Up<- inner_join(top_ROSI_Up,
                     rownames_to_column(coldata_ROSI, var ="samplename"),
                     by ="samplename")
head(top_ROSI_Up)
top_ROSI_up_anno <- data.frame(top_ROSI_Up)
top_ROSI_up_anno$symbol<-                                                                                      mapIds(edb,keys=top_ROSI_Up$ensgene,column="SYMBOL",keytype="GENEID",multiVals="first")
top_ROSI_up_anno$symbol

ggplot(top_ROSI_up_anno)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top Upregulated Genes ROSI vs DMSO")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))
        
####Down regulated DEGs####

dim(res_sig_normalized_counts_ROSI_Down) #156 27  

top_ROSI_Down <- data.frame(res_sig_normalized_counts_ROSI_Down[1:50,]) %>% rownames_to_column(var ="ensgene")
dim(top_ROSI_Down)#20 28
head(top_ROSI_Down)
top_ROSI_Down <- gather(top_ROSI_Down, key ="samplename", value ="normalized_counts",2:28)
head(top_ROSI_Down)

top_ROSI_Down<- inner_join(top_ROSI_Down,
                     rownames_to_column(coldata_ROSI, var ="samplename"),
                     by ="samplename")
head(top_ROSI_Down)
top_ROSI_Down_anno <- data.frame(top_ROSI_Down)
dim(top_ROSI_Down)
top_ROSI_Down_anno$symbol<-                                                                                      mapIds(edb,keys=top_ROSI_Down$ensgene,column="SYMBOL",keytype="GENEID",multiVals="first")
top_ROSI_Down_anno$symbol

ggplot(top_ROSI_Down_anno)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top Downregulated Genes ROSI vs DMSO")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))


```
###GSEA ofr No MSC data
```{r GSEA noMSC ROSI}
###GO analysis####
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(ggplot2)
# we use ggplot2 to add x axis labels (ex: ridgeplot)
head(ROSI_padj0.05_DEGS_anno)
ROSI_lfc <- ROSI_padj0.05_DEGS_anno$log2FoldChange
names(ROSI_lfc) <- ROSI_padj0.05_DEGS_anno$symbol
ROSI_lfc<-na.omit(ROSI_lfc)
# sort the list in decreasing order (required for clusterProfiler)
ROSI_lfc = sort(ROSI_lfc, decreasing = TRUE)
keytypes(org.Hs.eg.db)
head(ROSI_lfc)

#gene set enrichment
gse_ROSI_lfc <- gseGO(geneList=ROSI_lfc, 
             ont = "MF", 
             keyType = "SYMBOL", 
             minGSSize = 3, 
             maxGSSize = 300, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")
gse_ROSI_lfc
dotplot(gse_ROSI_lfc,showCategory=10) + ggtitle("GSEA ROSI vs DMSO No MSC")
require(DOSE)
dotplot(gse_ROSI_lfc, showCategory=10, split=".sign") + facet_grid(.~.sign)
###UP regulated###
ROSI_lfc_up <- ROSI_lfc[ROSI_lfc>0]
length(ROSI_lfc_up)#108 up regulated genes
gse_ROSI_lfc_Up <- gseGO(geneList=ROSI_lfc_up, 
             ont = "BP", 
             keyType = "SYMBOL", 
             minGSSize = 3, 
             maxGSSize = 300, 
             pvalueCutoff = 0.05,
             nPermSimple= 10000,
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")
write.csv(gse_ROSI_lfc_Up,"~/Documents/Git/MSC_Bulk/RNA_seq_Data/gse_ROSI_lfc_noMSC_Up.csv", quote = F)
dotplot(gse_ROSI_lfc_Up,showCategory=10) + ggtitle("GSEA ROSI vs DMSO noMSC Upregulated")
###EnrichGO
enrichGO_ROSI_lfc <- enrichGO(gene=unlist(names(ROSI_lfc_D3)), 
             ont = "BP", 
             keyType = "SYMBOL", 
             minGSSize = 3, 
             maxGSSize = 300, 
             pvalueCutoff = 0.05, 
             readable = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")
enrichGO_ROSI_lfc
#### Reduce clusters######
library(rrvgo)
simMatrix_ROSI <- calculateSimMatrix(gse_ROSI_lfc@result$ID, orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")

scores_simMatrix_ROSI <- setNames(-log10(gse_ROSI_lfc@result$qvalues), gse_ROSI_lfc@result$ID)
reducedTerms_ROSI <- reduceSimMatrix(simMatrix_ROSI, scores_simMatrix_ROSI, threshold = 0.7, orgdb = "org.Hs.eg.db")
#heatmapPlot(simMatrix_ROSI_D3,reducedTerms_ROSI_D3,annotateParent = T,annotationLabel = "parentTerm",fontsize= 6) # this is super noisy
scatterPlot(simMatrix_ROSI,reducedTerms_ROSI)
treemapPlot(reducedTerms_ROSI)
```
###Ploting DEGs Volcano
```{r Volcano Plots for ROSI no MSC}
library(ggrepel)
res_ROSI_anno <- data.frame(res_ROSI_noMSC_anno) %>% mutate(threshold = padj < 0.05)
res_ROSIpad.1_anno <- data.frame(res_order_ROSI) %>% mutate(threshold = padj < 0.05)res_order_ROSI
#Creating Volcano plot based on threshold of padjusted bvvalues
# ggplot(res_ROSI_anno)+
#   geom_point(aes(x = log2FoldChange, y = -log10(padj), color = threshold))+
#   xlab("log2 fold change")+
#   ylab("-log10 adjusted p-value")+
#   geom_vline(xintercept=c(-0.6, 0.6), col="red") +
#   geom_hline(yintercept=-log10(0.05), col="red")+
#   theme(legend.position ="none",
#         plot.title = element_text(size = rel(1.5), hjust =0.5),
#         axis.title = element_text(size = rel(1.25)))
# # creating differenitally expressed category
# res_ROSI_anno$diffexpressed <- "NO"
# # if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP" 
# res_ROSI_anno$diffexpressed[res_ROSI_anno$log2FoldChange > 1 & res_ROSI_anno$padj < 0.05] <- "UP"
# # if log2Foldchange < -0.6 and pvalue < 0.05, set as "DOWN"
# res_ROSI_anno$diffexpressed[res_ROSI_anno$log2FoldChange < -1 & res_ROSI_anno$padj < 0.05] <- "DOWN"
# sum(res_ROSI_anno$diffexpressed=="UP") #3169 up regulated genes
# sum(res_ROSI_anno$diffexpressed=="DOWN")#3429 down regulated genes
# #trying chaning the color based on the diffexpressed column  just added
# 
# p <- ggplot(res_ROSI_anno)+
#   geom_point(aes(x = log2FoldChange, y = -log10(padj), color = diffexpressed))+
#   xlab("log2 fold change")+
#   ylab("-log10 adjusted p-value")+
#   geom_vline(xintercept=c(-0.6, 0.6), col="red") +
#   geom_hline(yintercept=-log10(0.05), col="red")+
#   theme(legend.position ="right",
#         plot.title = element_text(size = rel(1.5), hjust =0.5),
#         axis.title = element_text(size = rel(1.25)))
# p
# ##changing the colors 
# mycolors <- c("blue", "red", "black")
# names(mycolors) <- c("DOWN", "UP", "NO")
# p2 <- p + scale_colour_manual(values = mycolors)
# p2

#Try the enhanced volcano plot with 
#this is the test code from the website
library(EnhancedVolcano)
head(res_ROSI_anno)#day 014 comparision annotated
EnhancedVolcano(res_ROSI_anno,
    lab = res_ROSI_anno$symbol,
    x = 'log2FoldChange',
    y = 'padj',
    pCutoff = 5e-2,
    FCcutoff = 0.25,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    title = "res_ROSI_anno no MSC",
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 16,
    legendIconSize = 5.0)
```
##Plot genes over Time Regressed
```{r Time}
# get the counts for the gen with the lowest p value over each time point
SD_Time <- plotCounts(DES_SD_NoExtra, which.min(res_SD_NoExtra$padj), 
                   intgroup = c("Day","condition"), returnData = TRUE)
#results(res_SD_NoExtra)

#res_SD_NoExtra
#Plot the most sig DEG over time
SD_Time$Day <- as.numeric(as.character(SD_Time$Day))
ggplot(SD_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_point() + stat_summary(fun.y=mean, geom="line") +
  scale_y_log10()
```
##Day by Day comparison
###D3
```{r Day 3 comparison}
#BiocManager::install("DESeq2")
suppressPackageStartupMessages(library("DESeq2"))

#Subsetting the data to only have the pertinent information
# cts_mat_ROSI<- cts_mat[,which(substr(colnames(cts_mat),2,2)=="R"|substr(colnames(cts_mat),2,2)=="D")]
# rownames(cts_mat_ROSI) <- substr(rownames(cts_mat_ROSI),1,15)#Removing the .# on the gene names
# coldata_ROSI <- coldata_norm[which(substr(rownames(coldata_norm),2,2)=="R"|substr(rownames(coldata_norm),2,2)=="D"),]
# dim(cts_mat_ROSI) #58037 27
# dim(coldata_ROSI) #27 4
cts_mat_ROSI_D3 <- cts_mat_ROSI[,1:7]
coldata_ROSI_D3 <- coldata_ROSI[1:7,]
##Creating DDS object from the subset data                                  
dds_ROSI_D3 <- DESeqDataSetFromMatrix(countData = cts_mat_ROSI_D3,
                                                 colData = coldata_ROSI_D3,
                                                 design = ~ condition                                               )
summary(dds_ROSI_D3)
head(dds_ROSI_D3)   
###Adding meta data columns
rownames(dds_ROSI_D3) <- substr(rownames(dds_ROSI_D3),1,15)
featureData <- data.frame(gene=substr(rownames(cts_mat_ROSI_D3),1,15))
mcols(dds_ROSI_D3) <- DataFrame(mcols(dds_ROSI_D3), featureData)
mcols(dds_ROSI_D3)
featureData$symbol <- cts_1021$SYMBOL# Adding mcol symbol data
mcols(dds_ROSI_D3) <- DataFrame(mcols(dds_ROSI_D3), featureData)
mcols(dds_ROSI_D3)
###Removing low quality rows
keep <- rowSums(counts(dds_ROSI_D3)) >= 10 #30255 genes passed this low bar
dds_ROSI_D3 <- dds_ROSI_D3[keep,]
####Making things factors
dds_ROSI_D3$condition <- factor(dds_ROSI_D3$condition, levels = c("control","ROSI"))
dds_ROSI_D3$condition
# Determine the size factors to use for normalization
dds_ROSI_D3<- estimateSizeFactors(dds_ROSI_D3)
sizeFactors(dds_ROSI_D3)
# Extract the normalized counts
normalized_counts_ROSI_D3 <- counts(dds_ROSI_D3, normalized=T)
head(normalized_counts_ROSI_D3)

# Transform the normalized counts 
vsd_ROSI_D3 <- vst(dds_ROSI_D3,blind=F )

# Extract the matrix of transformed counts
vsd_mat_ROSI_D3 <- assay(dds_ROSI_D3)

# Compute the correlation values between samples
vsd_cor_ROSI_D3 <- cor(vsd_mat_ROSI_D3) 

# Plot the heatmap
pheatmap(vsd_cor_ROSI_D3)

colData(vsd_ROSI_D3)
###Plot PCAs####
plotPCA(vsd_ROSI_D3, intgroup= "condition")

plotPCA(vsd_ROSI_D3, intgroup = c("Time","condition"))

DES_ROSI_D3 <- DESeq(dds_ROSI_D3)
res_ROSI_D3 <- results(DES_ROSI_D3)
resultsNames(res_ROSI_D3)
head(res_ROSI_D3)
mcols(res_ROSI_D3)
#Increase Alpha threshold
res_ROSI_D3_contrast <- results(DES_ROSI_D3, contrast=c("condition","ROSI","control"),alpha = 0.05)
head(res_ROSI_D3_contrast)

####DEG L2fc threshold alteration########
# #BiocManager::install("apeglm")
# #library(apeglm)
# # Shrink the log2 fold change estimates to be more accurate
# res_contr_ROSI_lfc_D3 <- results(DES_ROSI_D3, 
#                         contrast=c("condition","ROSI","control"),
#                         alpha = 0.05,
#                         lfcThreshold = 0.2)
# resultsNames(dds_ROSI_D3)
# 
# resLFC_ROSI_D3 <- lfcShrink(dds_ROSI_D3,
#                     contrast = c("condition","ROSI","control"),
#                     res = res_contr_ROSI_lfc_D3)
# summary(res_contr_ROSI_lfc_D3)
# #Save results as a dataframe
# res_cont_ROSI_D3_lfc.2 <- data.frame(res_contr_ROSI_lfc_D3)
#res_Ordered <- res_cont_ROSI[order(res_cont_ROSI$pvalue),]
#summary(res_Ordered)

### Subsetting for padj and log2fc data ####
sum(res_ROSI_D3_contrast$padj < 0.05, na.rm=TRUE) #382 lowering the Padjusted value removed some of the genes
res_ROSI_D3_contrast_padj_0.05 <- res_ROSI_D3_contrast[which(res_ROSI_D3_contrast$padj<0.05),]


#All significant normalized counts table
res_order_ROSI_D3 <-res_ROSI_D3_contrast_padj_0.05[order(res_ROSI_D3_contrast_padj_0.05$padj),]

res_sig_order_ROSIcontrD3_padj_0.05 <- res_ROSI_D3_contrast_padj_0.05[order(res_ROSI_D3_contrast_padj_0.05$padj),]
dim(res_order_ROSI_D3)#382 6


##Annnotate significant table
suppressPackageStartupMessages(library(org.Hs.eg.db))
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(EnsDb.Hsapiens.v86)
edb <- EnsDb.Hsapiens.v86
organism(edb)
keytypes(edb)
keys(edb,keytype = "GENEID")
###Annotate results table#####
res_ROSI_contrD3_anno <- data.frame(res_ROSI_D3_contrast_padj_0.05)
res_ROSI_contrD3_anno$symbol <- mapIds(edb,keys=rownames(res_ROSI_contrD3_anno),column="SYMBOL",keytype="GENEID",multiVals="first")
dim(res_ROSI_contrD3_anno)
###Write and save data###
library(writexl)
#writexl::write_xlsx(res_ROSI_contrD3_anno,"~/Documents/Git/MSC_Bulk/RNA_seq_Data/ROSI_noMSC_DEGs_D3_sig_anno.xlsx")

```
###D3 Visualization
```{r D3 vs D3 visualizations of samples only results}
# trying to run contrast on the contrasted data

head(res_ROSI_contrD3_anno)

## Subsetting significant counts
sum(res_ROSI_contrD3_anno$padj < 0.05, na.rm=TRUE) #10706 A lot more significant genes from day 7 to day 14
res_ROSI_contrD3_anno_padj_0.05 <- res_ROSI_contrD3_anno[which(res_ROSI_contrD3_anno$padj<0.05),]
###l2fc cutoff
res_ROSI_contrD3_anno_padj_0.05_l2fc1 <- res_ROSI_contrD3_anno_padj_0.05[which(abs(res_ROSI_contrD3_anno_padj_0.05$log2FoldChange)>1),]
dim(res_ROSI_contrD3_anno_padj_0.05_l2fc1)#25 7
#Data to use: 
#           res_ROSI_anno(all results annotated with threshold cutoff)
#normalized counts tables:
#           vsd_mat_ROSI(this is the normalized data)
#           res_sig_normalized_counts_ROSI(normalized data subset form the standard constrast)

res_ROSI_contrD3_anno_padj_0.05_NC <- vsd_mat_ROSI_D3[rownames(res_ROSI_contrD3_anno_padj_0.05),]
res_ROSI_contrD3_anno_padj_0.05_l2fc_NC <- vsd_mat_ROSI_D3[rownames(res_ROSI_contrD3_anno_padj_0.05_l2fc1),]
####Make the normalized counts from day contrast#####
heaed(res_ROSI_contrD3_anno_padj_0.05)
head(res_ROSI_contrD3_anno_padj_0.05_NC)

##Heatmap####
library(pheatmap)
library(ComplexHeatmap)


pheatmap(res_ROSI_contrD3_anno_padj_0.05_NC,
         cluster_rows =T,
         color = heat_colors,
         cluster_cols = F,
         show_rownames =F,
         annotation = select(coldata_ROSI, condition, Day),
         scale = "row")
colnames(res_sig_normalized_counts_ROSI)

####Volcano#######
##Annnotate significant table
suppressPackageStartupMessages(library(org.Hs.eg.db))
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(EnsDb.Hsapiens.v86)
edb <- EnsDb.Hsapiens.v86
organism(edb)
keytypes(edb)
keys(edb,keytype = "GENEID")
###Annotate results table#####
res_ROSI_D3_contrast_anno <- data.frame(res_ROSI_D3_contrast)
res_ROSI_D3_contrast_anno$symbol <- mapIds(edb,keys=rownames(res_ROSI_D3_contrast),column="SYMBOL",keytype="GENEID",multiVals="first")
###Volcano Plot ######

EnhancedVolcano(res_ROSI_D3_contrast_anno,
    lab = res_ROSI_D3_contrast_anno$symbol,
    x = 'log2FoldChange',
    y = 'padj',
    pCutoff = 5e-2,
    FCcutoff = .75,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 6.0,
    title = "results ROSI vs DMSO contrast day 7",
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 4.0,
    drawConnectors = T,
    widthConnectors = 0.75,
    max.overlaps = 50)
###GO analysis####
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(ggplot2)
# we use ggplot2 to add x axis labels (ex: ridgeplot)
head(res_ROSI_contrD3_anno_padj_0.05)
ROSI_lfc_D3 <- res_ROSI_contrD3_anno_padj_0.05$log2FoldChange
names(ROSI_lfc_D3) <- res_ROSI_contrD3_anno_padj_0.05$symbol
ROSI_lfc_D3<-na.omit(ROSI_lfc_D3)
# sort the list in decreasing order (required for clusterProfiler)
ROSI_lfc_D3 = sort(ROSI_lfc_D3, decreasing = TRUE)
keytypes(org.Hs.eg.db)
head(ROSI_lfc_D3)

#gene set enrichment
gse_ROSI_lfc_D3 <- gseGO(geneList=ROSI_lfc_D3, 
             ont = "MF", 
             keyType = "SYMBOL", 
             minGSSize = 3, 
             maxGSSize = 300, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")
gse_ROSI_lfc_D3
dotplot(gse_ROSI_lfc_D3,showCategory=10) + ggtitle("GSEA ROSI vs DMSO D3")
dotplot(gse_ROSI_lfc_D3, showCategory=10, split=".sign") + facet_grid(.~.sign) + ggtitle("GSEA ROSI vs DMSO D3")
###UP regulated###
ROSI_lfc_D3_up <- ROSI_lfc_D3[ROSI_lfc_D3>0]
length(ROSI_lfc_D3_up)#108 up regulated genes
gse_ROSI_lfc_D3_Up <- gseGO(geneList=ROSI_lfc_D3_up, 
             ont = "BP", 
             keyType = "SYMBOL", 
             minGSSize = 3, 
             maxGSSize = 300, 
             pvalueCutoff = 0.05,
             nPermSimple= 10000,
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")
write.csv(gse_ROSI_lfc_D3_Up,"~/Documents/Git/MSC_Bulk/RNA_seq_Data/gse_ROSI_lfc_D3_Up.csv", quote = F)
dotplot(gse_ROSI_lfc_D3_Up,showCategory=10) + ggtitle("GSEA ROSI vs DMSO D3 Upregulated")
###EnrichGO
enrichGO_ROSI_lfc_D3 <- enrichGO(gene=unlist(names(ROSI_lfc_D3)), 
             ont = "BP", 
             keyType = "SYMBOL", 
             minGSSize = 3, 
             maxGSSize = 300, 
             pvalueCutoff = 0.05, 
             readable = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")
enrichGO_ROSI_lfc_D3
#### Reduce clusters######
library(rrvgo)
simMatrix_ROSI_D3 <- calculateSimMatrix(gse_ROSI_lfc_D3@result$ID, orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")

scores_simMatrix_ROSI_D3 <- setNames(-log10(gse_ROSI_lfc_D3@result$qvalues), gse_ROSI_lfc_D3@result$ID)
reducedTerms_ROSI_D3 <- reduceSimMatrix(simMatrix_ROSI_D3, scores_simMatrix_ROSI_D3, threshold = 0.7, orgdb = "org.Hs.eg.db")
#heatmapPlot(simMatrix_ROSI_D3,reducedTerms_ROSI_D3,annotateParent = T,annotationLabel = "parentTerm",fontsize= 6) # this is super noisy
scatterPlot(simMatrix_ROSI_D3,reducedTerms_ROSI_D3)
treemapPlot(reducedTerms_ROSI_D3)

### Seperating up vs down regulated genes ####
####Up Regulated DEGS####
res_ROSI_contrD3_anno_padj_0.05 <- res_ROSI_contrD3_anno_padj_0.05[order(res_ROSI_contrD3_anno_padj_0.05$padj),]
head(res_ROSI_contrD3_anno_padj_0.05_NC)
#normalized counts upregulated
res_ROSI_contrD3_anno_padj_0.05_NC_UP <- res_ROSI_contrD3_anno_padj_0.05_NC[rownames(res_ROSI_contrD3_anno_padj_0.05[which(res_ROSI_contrD3_anno_padj_0.05$log2FoldChange>0),]),]
dim(res_ROSI_contrD3_anno_padj_0.05_NC_UP) #108 7
top_ROSI_D3_Up <- data.frame(res_ROSI_contrD3_anno_padj_0.05_NC_UP) %>% rownames_to_column(var ="ensgene")
dim(top_ROSI_D3_Up)
head(top_ROSI_D3_Up)
top_ROSI_D3_Up <- gather(top_ROSI_D3_Up, key ="samplename", value ="normalized_counts",2:8)
head(top_ROSI_D3_Up)

top_ROSI_D3_Up<- inner_join(top_ROSI_D3_Up,
                     rownames_to_column(coldata_ROSI, var ="samplename"),
                     by ="samplename")
head(top_ROSI_D3_Up)
top_ROSI_up_D3_anno <- data.frame(top_ROSI_D3_Up)
top_ROSI_up_D3_anno$symbol<-                                                                                      mapIds(edb,keys=top_ROSI_D3_Up$ensgene,column="SYMBOL",keytype="GENEID",multiVals="first")
top_ROSI_up_D3_anno$symbol

ggplot(top_ROSI_up_D3_anno)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top Upregulated Genes ROSI vs DMSO Day 3")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))
        
####Down regulated DEGs####
res_ROSI_contrD3_anno_padj_0.05 <- res_ROSI_contrD3_anno_padj_0.05[order(res_ROSI_contrD3_anno_padj_0.05$padj),]
head(res_ROSI_contrD3_anno_padj_0.05_NC)
#normalized counts upregulated
res_ROSI_contrD3_anno_padj_0.05_NC_Down <- res_ROSI_contrD3_anno_padj_0.05_NC[rownames(res_ROSI_contrD3_anno_padj_0.05[which(res_ROSI_contrD3_anno_padj_0.05$log2FoldChange<0),]),]
dim(res_ROSI_contrD7_anno_padj_0.05_NC_Down) #108 7
#dim(res_sig_normalized_counts_ROSI_Down) #156 27  

top_ROSI_D3_Down <- data.frame(res_ROSI_contrD7_anno_padj_0.05_NC_Down[1:50,]) %>% rownames_to_column(var ="ensgene")
dim(top_ROSI_D3_Down)#20 28
head(top_ROSI_D3_Down)
top_ROSI_D3_Down <- gather(top_ROSI_D3_Down, key ="samplename", value ="normalized_counts",2:28)
head(top_ROSI_D3_Down)

top_ROSI_D3_Down<- inner_join(top_ROSI_D3_Down,
                     rownames_to_column(coldata_ROSI, var ="samplename"),
                     by ="samplename")
head(top_ROSI_D3_Down)
top_ROSI_Down_anno <- data.frame(top_ROSI_D3_Down)
dim(top_ROSI_D3_Down)
top_ROSI_D3_Down_anno$symbol<-                                                                                      mapIds(edb,keys=top_ROSI_D3_Down$ensgene,column="SYMBOL",keytype="GENEID",multiVals="first")
top_ROSI_D3_Down_anno$symbol

ggplot(top_ROSI_D3_Down_anno)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top Downregulated Genes ROSI vs DMSO Day 3")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))
```
###D7
```{r Day 7 comparison}
#BiocManager::install("DESeq2")
suppressPackageStartupMessages(library("DESeq2"))

#Subsetting the data to only have the pertinent information
# cts_mat_ROSI<- cts_mat[,which(substr(colnames(cts_mat),2,2)=="R"|substr(colnames(cts_mat),2,2)=="D")]
# rownames(cts_mat_ROSI) <- substr(rownames(cts_mat_ROSI),1,15)#Removing the .# on the gene names
# coldata_ROSI <- coldata_norm[which(substr(rownames(coldata_norm),2,2)=="R"|substr(rownames(coldata_norm),2,2)=="D"),]
colnames(cts_mat_ROSI) #58037 27
# dim(coldata_ROSI) #27 4
cts_mat_ROSI_D7 <- cts_mat_ROSI[,8:15]
coldata_ROSI_D7 <- coldata_ROSI[8:15,]
##Creating DDS object from the subset data                                  
dds_ROSI_D7 <- DESeqDataSetFromMatrix(countData = cts_mat_ROSI_D7,
                                                 colData = coldata_ROSI_D7,
                                                 design = ~ condition                                               )
summary(dds_ROSI_D7)
head(dds_ROSI_D7)   
###Adding meta data columns
rownames(dds_ROSI_D7) <- substr(rownames(dds_ROSI_D7),1,15)
featureData <- data.frame(gene=substr(rownames(cts_mat_ROSI_D7),1,15))
mcols(dds_ROSI_D7) <- DataFrame(mcols(dds_ROSI_D7), featureData)
mcols(dds_ROSI_D7)
featureData$symbol <- cts_1021$SYMBOL# Adding mcol symbol data
mcols(dds_ROSI_D7) <- DataFrame(mcols(dds_ROSI_D7), featureData)
mcols(dds_ROSI_D7)
###Removing low quality rows
keep <- rowSums(counts(dds_ROSI_D7)) >= 10 #30255 genes passed this low bar
dds_ROSI_D7 <- dds_ROSI_D7[keep,]
####Making things factors
dds_ROSI_D7$condition <- factor(dds_ROSI_D7$condition, levels = c("control","ROSI"))
dds_ROSI_D7$condition
# Determine the size factors to use for normalization
dds_ROSI_D7<- estimateSizeFactors(dds_ROSI_D7)
sizeFactors(dds_ROSI_D7)
# Extract the normalized counts
normalized_counts_ROSI_D7 <- counts(dds_ROSI_D7, normalized=T)
head(normalized_counts_ROSI_D7)

# Transform the normalized counts 
vsd_ROSI_D7 <- vst(dds_ROSI_D7,blind=F )

# Extract the matrix of transformed counts
vsd_mat_ROSI_D7 <- assay(dds_ROSI_D7)

# Compute the correlation values between samples
vsd_cor_ROSI_D7 <- cor(vsd_mat_ROSI_D7) 

# Plot the heatmap
pheatmap(vsd_cor_ROSI_D7)

colData(vsd_ROSI_D7)
###Plot PCAs####
plotPCA(vsd_ROSI_D7, intgroup= "condition")

plotPCA(vsd_ROSI_D7, intgroup = c("Time","condition"))

DES_ROSI_D7 <- DESeq(dds_ROSI_D7)
res_ROSI_D7 <- results(DES_ROSI_D7)
resultsNames(res_ROSI_D7)
head(res_ROSI_D7)
mcols(res_ROSI_D7)
#Increase Alpha threshold
res_ROSI_D7_contrast <- results(DES_ROSI_D7, contrast=c("condition","ROSI","control"),alpha = 0.05)
head(res_ROSI_D7_contrast)

####DEG L2fc threshold alteration########
# #BiocManager::install("apeglm")
# #library(apeglm)
# # Shrink the log2 fold change estimates to be more accurate
# res_contr_ROSI_lfc_D7 <- results(DES_ROSI_D7, 
#                         contrast=c("condition","ROSI","control"),
#                         alpha = 0.05,
#                         lfcThreshold = 0.2)
# resultsNames(dds_ROSI_D7)
# 
# resLFC_ROSI_D7 <- lfcShrink(dds_ROSI_D7,
#                     contrast = c("condition","ROSI","control"),
#                     res = res_contr_ROSI_lfc_D7)
# summary(res_contr_ROSI_lfc_D7)
# #Save results as a dataframe
# res_cont_ROSI_D7_lfc.2 <- data.frame(res_contr_ROSI_lfc_D7)
#res_Ordered <- res_cont_ROSI[order(res_cont_ROSI$pvalue),]
#summary(res_Ordered)

### Subsetting for padj and log2fc data ####
sum(res_ROSI_D7_contrast$padj < 0.05, na.rm=TRUE) #382 lowering the Padjusted value removed some of the genes
res_ROSI_D7_contrast_padj_0.05 <- res_ROSI_D7_contrast[which(res_ROSI_D7_contrast$padj<0.05),]


#All significant normalized counts table
res_order_ROSI_D7 <-res_ROSI_D7_contrast_padj_0.05[order(res_ROSI_D7_contrast_padj_0.05$padj),]

res_sig_order_ROSIcontrD7_padj_0.05 <- res_ROSI_D7_contrast_padj_0.05[order(res_ROSI_D7_contrast_padj_0.05$padj),]
dim(res_order_ROSI_D7)#382 6


##Annnotate significant table
suppressPackageStartupMessages(library(org.Hs.eg.db))
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(EnsDb.Hsapiens.v86)
edb <- EnsDb.Hsapiens.v86
organism(edb)
keytypes(edb)
keys(edb,keytype = "GENEID")
###Annotate results table#####
res_ROSI_contrD7_anno <- data.frame(res_ROSI_D7_contrast_padj_0.05)
res_ROSI_contrD7_anno$symbol <- mapIds(edb,keys=rownames(res_ROSI_contrD7_anno),column="SYMBOL",keytype="GENEID",multiVals="first")

###Write and save data###
library(writexl)
#writexl::write_xlsx(res_ROSI_contrD7_anno,"~/Documents/Git/MSC_Bulk/RNA_seq_Data/ROSI_noMSC_DEGs_D7_sig_anno.xlsx")

```
###D7 Visualization
```{r D7 samples only results}
# trying to run contrast on the contrasted data

head(res_ROSI_contrD7_anno)

## Subsetting significant counts
sum(res_ROSI_contrD7_anno$padj < 0.05, na.rm=TRUE) #21 
res_ROSI_contrD7_anno_padj_0.05 <- res_ROSI_contrD7_anno[which(res_ROSI_contrD7_anno$padj<0.05),]
###l2fc cutoff
res_ROSI_contrD7_anno_padj_0.05_l2fc1 <- res_ROSI_contrD7_anno_padj_0.05[which(abs(res_ROSI_contrD7_anno_padj_0.05$log2FoldChange)>1),]
dim(res_ROSI_contrD7_anno_padj_0.05_l2fc1)#0 7
#Data to use: 
#           res_ROSI_anno(all results annotated with threshold cutoff)
#normalized counts tables:
#           vsd_mat_ROSI(this is the normalized data)
#           res_sig_normalized_counts_ROSI(normalized data subset form the standard constrast)

res_ROSI_contrD7_anno_padj_0.05_NC <- vsd_mat_ROSI_D7[rownames(res_ROSI_contrD7_anno_padj_0.05),]
res_ROSI_contrD7_anno_padj_0.05_l2fc_NC <- vsd_mat_ROSI_D7[rownames(res_ROSI_contrD7_anno_padj_0.05_l2fc1),]
####Make the normalized counts from day contrast#####
head(res_ROSI_contrD7_anno_padj_0.05)
head(res_ROSI_contrD7_anno_padj_0.05_NC)
##Heatmap####
library(pheatmap)
library(ComplexHeatmap)


pheatmap(res_ROSI_contrD7_anno_padj_0.05_NC,
         cluster_rows =T,
         color = heat_colors,
         cluster_cols = F,
         show_rownames =F,
         annotation = select(coldata_ROSI, condition, Day),
         scale = "row")
colnames(res_sig_normalized_counts_ROSI)

####Volcano#######
##Annnotate significant table
suppressPackageStartupMessages(library(org.Hs.eg.db))
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(EnsDb.Hsapiens.v86)
edb <- EnsDb.Hsapiens.v86
organism(edb)
keytypes(edb)
keys(edb,keytype = "GENEID")
###Annotate results table#####
res_ROSI_D7_contrast_anno <- data.frame(res_ROSI_D7_contrast)
res_ROSI_D7_contrast_anno$symbol <- mapIds(edb,keys=rownames(res_ROSI_D7_contrast),column="SYMBOL",keytype="GENEID",multiVals="first")
###

EnhancedVolcano(res_ROSI_D7_contrast_anno,
    lab = res_ROSI_D7_contrast_anno$symbol,
    x = 'log2FoldChange',
    y = 'padj',
    pCutoff = 5e-2,
    FCcutoff = .75,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 6.0,
    title = "results ROSI vs DMSO contrast day 7",
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 4.0,
    drawConnectors = T,
    widthConnectors = 0.75,
    max.overlaps = 50)
###GO analysis####
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(ggplot2)
# we use ggplot2 to add x axis labels (ex: ridgeplot)
head(res_ROSI_contrD7_anno_padj_0.05)
ROSI_lfc_D7 <- res_ROSI_contrD7_anno_padj_0.05$log2FoldChange
names(ROSI_lfc_D7) <- res_ROSI_contrD7_anno_padj_0.05$symbol
ROSI_lfc_D7<-na.omit(ROSI_lfc_D7)
# sort the list in decreasing order (required for clusterProfiler)
ROSI_lfc_D7 = sort(ROSI_lfc_D7, decreasing = TRUE)
keytypes(org.Hs.eg.db)
head(ROSI_lfc_D7)

#gene set enrichment
gse_ROSI_lfc_D7 <- gseGO(geneList=ROSI_lfc_D7, 
             ont = "All", 
             keyType = "SYMBOL", 
             minGSSize = 3, 
             maxGSSize = 300, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")
gse_ROSI_lfc_D7
dotplot(gse_ROSI_lfc_D7,showCategory=10) + ggtitle("GSEA ROSI vs DMSO D7")
#### Reduce clusters######
library(rrvgo)
simMatrix_ROSI_D7 <- calculateSimMatrix(gse_ROSI_lfc_D7@result$ID, orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")

scores_simMatrix_ROSI_D7 <- setNames(-log10(gse_ROSI_lfc_D7@result$qvalues), gse_ROSI_lfc_D7@result$ID)
reducedTerms_ROSI_D7 <- reduceSimMatrix(simMatrix_ROSI_D7, scores_simMatrix_ROSI_D7, threshold = 0.7, orgdb = "org.Hs.eg.db")
#heatmapPlot(simMatrix_ROSI_D7,reducedTerms_ROSI_D7,annotateParent = T,annotationLabel = "parentTerm",fontsize= 6) # this is super noisy
scatterPlot(simMatrix_ROSI_D7,reducedTerms_ROSI_D7)
treemapPlot(reducedTerms_ROSI_D7)
###UP regulated###
ROSI_lfc_D7_up <- ROSI_lfc_D7[ROSI_lfc_D7>0]
length(ROSI_lfc_D7_up)#108 up regulated genes
gse_ROSI_lfc_D7_Up <- gseGO(geneList=ROSI_lfc_D7_up, 
             ont = "BP", 
             keyType = "SYMBOL", 
             minGSSize = 3, 
             maxGSSize = 300, 
             pvalueCutoff = 0.05,
             nPermSimple= 10000,
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")
write.csv(gse_ROSI_lfc_D7_Up,"~/Documents/Git/MSC_Bulk/RNA_seq_Data/gse_ROSI_lfc_D7_Up.csv", quote = F)
dotplot(gse_ROSI_lfc_D7_Up,showCategory=10) + ggtitle("GSEA ROSI vs DMSO D7 Upregulated")

### Seperating up vs down regulated genes ####
####Up Regulated DEGS####
res_ROSI_contrD7_anno_padj_0.05 <- res_ROSI_contrD7_anno_padj_0.05[order(res_ROSI_contrD7_anno_padj_0.05$padj),]
head(res_ROSI_contrD7_anno_padj_0.05_NC)
#normalized counts upregulated
res_ROSI_contrD7_anno_padj_0.05_NC_UP <- res_ROSI_contrD7_anno_padj_0.05_NC[rownames(res_ROSI_contrD7_anno_padj_0.05[which(res_ROSI_contrD7_anno_padj_0.05$log2FoldChange>0),]),]
dim(res_ROSI_contrD7_anno_padj_0.05_NC_UP) #6 8
top_ROSI_D7_Up <- data.frame(res_ROSI_contrD7_anno_padj_0.05_NC_UP) %>% rownames_to_column(var ="ensgene")
dim(top_ROSI_D7_Up) # 6 9 
head(top_ROSI_D7_Up)
top_ROSI_D7_Up <- gather(top_ROSI_D7_Up, key ="samplename", value ="normalized_counts",2:8)
head(top_ROSI_D7_Up)

top_ROSI_D7_Up<- inner_join(top_ROSI_D7_Up,
                     rownames_to_column(coldata_ROSI, var ="samplename"),
                     by ="samplename")
head(top_ROSI_D7_Up)
top_ROSI_up_D7_anno <- data.frame(top_ROSI_D7_Up)
top_ROSI_up_D7_anno$symbol<-                                                                                      mapIds(edb,keys=top_ROSI_D7_Up$ensgene,column="SYMBOL",keytype="GENEID",multiVals="first")
top_ROSI_up_D7_anno$symbol

ggplot(top_ROSI_up_D7_anno)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top Upregulated Genes ROSI vs DMSO Day 3")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))
        
####Down regulated DEGs####
res_ROSI_contrD7_anno_padj_0.05 <- res_ROSI_contrD7_anno_padj_0.05[order(res_ROSI_contrD7_anno_padj_0.05$padj),]
head(res_ROSI_contrD7_anno_padj_0.05_NC)
#normalized counts upregulated
res_ROSI_contrD7_anno_padj_0.05_NC_Down <- res_ROSI_contrD7_anno_padj_0.05_NC[rownames(res_ROSI_contrD7_anno_padj_0.05[which(res_ROSI_contrD7_anno_padj_0.05$log2FoldChange<0),]),]
dim(res_ROSI_contrD7_anno_padj_0.05_NC_Down) #108 7
#dim(res_sig_normalized_counts_ROSI_Down) #156 27  

top_ROSI_D7_Down <- data.frame(res_ROSI_contrD7_anno_padj_0.05_NC_Down[1:50,]) %>% rownames_to_column(var ="ensgene")
dim(top_ROSI_D7_Down)#20 28
head(top_ROSI_D7_Down)
top_ROSI_D7_Down <- gather(top_ROSI_D7_Down, key ="samplename", value ="normalized_counts",2:28)
head(top_ROSI_D7_Down)

top_ROSI_D7_Down<- inner_join(top_ROSI_D7_Down,
                     rownames_to_column(coldata_ROSI, var ="samplename"),
                     by ="samplename")
head(top_ROSI_D7_Down)
top_ROSI_Down_anno <- data.frame(top_ROSI_D7_Down)
dim(top_ROSI_D7_Down)
top_ROSI_D7_Down_anno$symbol<-                                                                                      mapIds(edb,keys=top_ROSI_D7_Down$ensgene,column="SYMBOL",keytype="GENEID",multiVals="first")
top_ROSI_D7_Down_anno$symbol

ggplot(top_ROSI_D7_Down_anno)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top Downregulated Genes ROSI vs DMSO Day 3")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))
```
###D10
```{r Day 10 comparison}
#BiocManager::install("DESeq2")
suppressPackageStartupMessages(library("DESeq2"))

#Subsetting the data to only have the pertinent information
# cts_mat_ROSI<- cts_mat[,which(substr(colnames(cts_mat),2,2)=="R"|substr(colnames(cts_mat),2,2)=="D")]
# rownames(cts_mat_ROSI) <- substr(rownames(cts_mat_ROSI),1,15)#Removing the .# on the gene names
# coldata_ROSI <- coldata_norm[which(substr(rownames(coldata_norm),2,2)=="R"|substr(rownames(coldata_norm),2,2)=="D"),]
colnames(cts_mat_ROSI) #58037 27
# dim(coldata_ROSI) #27 4
cts_mat_ROSI_D10 <- cts_mat_ROSI[,16:21]
coldata_ROSI_D10 <- coldata_ROSI[16:21,]
##Creating DDS object from the subset data                                  
dds_ROSI_D10 <- DESeqDataSetFromMatrix(countData = cts_mat_ROSI_D10,
                                                 colData = coldata_ROSI_D10,
                                                 design = ~ condition                                               )
summary(dds_ROSI_D10)
head(dds_ROSI_D10)   
###Adding meta data columns
rownames(dds_ROSI_D10) <- substr(rownames(dds_ROSI_D10),1,15)
featureData <- data.frame(gene=substr(rownames(cts_mat_ROSI_D10),1,15))
mcols(dds_ROSI_D10) <- DataFrame(mcols(dds_ROSI_D10), featureData)
mcols(dds_ROSI_D10)
featureData$symbol <- cts_1021$SYMBOL# Adding mcol symbol data
mcols(dds_ROSI_D10) <- DataFrame(mcols(dds_ROSI_D10), featureData)
mcols(dds_ROSI_D10)
###Removing low quality rows
keep <- rowSums(counts(dds_ROSI_D10)) >= 10 #30255 genes passed this low bar
dds_ROSI_D10 <- dds_ROSI_D10[keep,]
####Making things factors
dds_ROSI_D10$condition <- factor(dds_ROSI_D10$condition, levels = c("control","ROSI"))
dds_ROSI_D10$condition
# Determine the size factors to use for normalization
dds_ROSI_D10<- estimateSizeFactors(dds_ROSI_D10)
sizeFactors(dds_ROSI_D10)
# Extract the normalized counts
normalized_counts_ROSI_D10 <- counts(dds_ROSI_D10, normalized=T)
head(normalized_counts_ROSI_D10)

# Transform the normalized counts 
vsd_ROSI_D10 <- vst(dds_ROSI_D10,blind=F )

# Extract the matrix of transformed counts
vsd_mat_ROSI_D10 <- assay(dds_ROSI_D10)

# Compute the correlation values between samples
vsd_cor_ROSI_D10 <- cor(vsd_mat_ROSI_D10) 

# Plot the heatmap
pheatmap(vsd_cor_ROSI_D10)

colData(vsd_ROSI_D10)
###Plot PCAs####
plotPCA(vsd_ROSI_D10, intgroup= "condition")

plotPCA(vsd_ROSI_D10, intgroup = c("Time","condition"))

DES_ROSI_D10 <- DESeq(dds_ROSI_D10)
res_ROSI_D10 <- results(DES_ROSI_D10)
resultsNames(res_ROSI_D10)
head(res_ROSI_D10)
mcols(res_ROSI_D10)
#Increase Alpha threshold
res_ROSI_D10_contrast <- results(DES_ROSI_D10, contrast=c("condition","ROSI","control"),alpha = 0.05)
head(res_ROSI_D10_contrast)

####DEG L2fc threshold alteration########
# #BiocManager::install("apeglm")
# #library(apeglm)
# # Shrink the log2 fold change estimates to be more accurate
# res_contr_ROSI_lfc_D10 <- results(DES_ROSI_D10, 
#                         contrast=c("condition","ROSI","control"),
#                         alpha = 0.05,
#                         lfcThreshold = 0.2)
# resultsNames(dds_ROSI_D10)
# 
# resLFC_ROSI_D10 <- lfcShrink(dds_ROSI_D10,
#                     contrast = c("condition","ROSI","control"),
#                     res = res_contr_ROSI_lfc_D10)
# summary(res_contr_ROSI_lfc_D10)
# #Save results as a dataframe
# res_cont_ROSI_D10_lfc.2 <- data.frame(res_contr_ROSI_lfc_D10)
#res_Ordered <- res_cont_ROSI[order(res_cont_ROSI$pvalue),]
#summary(res_Ordered)

### Subsetting for padj and log2fc data ####
sum(res_ROSI_D10_contrast$padj < 0.05, na.rm=TRUE) #973 lowering the Padjusted value removed some of the genes
res_ROSI_D10_contrast_padj_0.05 <- res_ROSI_D10_contrast[which(res_ROSI_D10_contrast$padj<0.05),]


#All significant normalized counts table
res_order_ROSI_D10 <-res_ROSI_D10_contrast_padj_0.05[order(res_ROSI_D10_contrast_padj_0.05$padj),]

res_sig_order_ROSIcontrD10_padj_0.05 <- res_ROSI_D10_contrast_padj_0.05[order(res_ROSI_D10_contrast_padj_0.05$padj),]
dim(res_order_ROSI_D10)#973 6


##Annnotate significant table
suppressPackageStartupMessages(library(org.Hs.eg.db))
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(EnsDb.Hsapiens.v86)
edb <- EnsDb.Hsapiens.v86
organism(edb)
keytypes(edb)
keys(edb,keytype = "GENEID")
###Annotate results table#####
res_ROSI_contrD10_anno <- data.frame(res_order_ROSI_D10)
res_ROSI_contrD10_anno$symbol <- mapIds(edb,keys=rownames(res_ROSI_contrD10_anno),column="SYMBOL",keytype="GENEID",multiVals="first")

###Write and save data###
library(writexl)
writexl::write_xlsx(res_ROSI_contrD10_anno,"~/Documents/Git/MSC_Bulk/RNA_seq_Data/ROSI_noMSC_DEGs_D10_sig_anno.xlsx")
```
###D10 Visualization
```{r D10 samples only results}
# trying to run contrast on the contrasted data

head(res_ROSI_contrD10_anno)

## Subsetting significant counts
sum(res_ROSI_contrD10_anno$padj < 0.05, na.rm=TRUE) #973 A lot more significant genes from day 7 to day 14
res_ROSI_contrD10_anno_padj_0.05 <- res_ROSI_contrD10_anno[which(res_ROSI_contrD10_anno$padj<0.05),]
###l2fc cutoff
res_ROSI_contrD10_anno_padj_0.05_l2fc1 <- res_ROSI_contrD10_anno_padj_0.05[which(abs(res_ROSI_contrD10_anno_padj_0.05$log2FoldChange)>1),]
dim(res_ROSI_contrD10_anno_padj_0.05_l2fc1)#289 7
#Data to use: 
#           res_ROSI_anno(all results annotated with threshold cutoff)
#normalized counts tables:
#           vsd_mat_ROSI(this is the normalized data)
#           res_sig_normalized_counts_ROSI(normalized data subset form the standard constrast)

res_ROSI_contrD10_padj_0.05_NC <- vsd_mat_ROSI_D10[rownames(res_ROSI_contrD10_anno_padj_0.05),]
res_ROSI_contrD10_padj_0.05_l2fc_NC <- vsd_mat_ROSI_D10[rownames(res_ROSI_contrD10_anno_padj_0.05_l2fc1),]
####Make the normalized counts from day contrast#####
head(res_ROSI_contrD10_anno_padj_0.05)
head(res_ROSI_contrD10_padj_0.05_NC)
##Heatmap####
library(pheatmap)
library(ComplexHeatmap)


pheatmap(res_ROSI_contrD10_padj_0.05_NC,
         cluster_rows =T,
         color = heat_colors,
         cluster_cols = F,
         show_rownames =F,
         annotation = select(coldata_ROSI, condition, Day),
         scale = "row")


####Volcano#######
##Annnotate significant table
suppressPackageStartupMessages(library(org.Hs.eg.db))
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(EnsDb.Hsapiens.v86)
edb <- EnsDb.Hsapiens.v86
organism(edb)
keytypes(edb)
keys(edb,keytype = "GENEID")
###Annotate results table#####
res_ROSI_D10_contrast_anno <- data.frame(res_ROSI_D10_contrast)
res_ROSI_D10_contrast_anno$symbol <- mapIds(edb,keys=rownames(res_ROSI_D10_contrast),column="SYMBOL",keytype="GENEID",multiVals="first")
###

EnhancedVolcano(res_ROSI_D10_contrast_anno,
    lab = res_ROSI_D10_contrast_anno$symbol,
    x = 'log2FoldChange',
    y = 'padj',
    pCutoff = 5e-2,
    FCcutoff = 1,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 6.0,
    title = "results ROSI vs DMSO contrast day 10",
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 4.0,
    drawConnectors = T,
    widthConnectors = 0.75,
    max.overlaps = 20)
###GO analysis####
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(ggplot2)
# we use ggplot2 to add x axis labels (ex: ridgeplot)
head(res_ROSI_contrD10_anno_padj_0.05)
ROSI_lfc_D10 <- res_ROSI_contrD10_anno_padj_0.05$log2FoldChange
names(ROSI_lfc_D10) <- res_ROSI_contrD10_anno_padj_0.05$symbol
ROSI_lfc_D10<-na.omit(ROSI_lfc_D10)
# sort the list in decreasing order (required for clusterProfiler)
ROSI_lfc_D10 = sort(ROSI_lfc_D10, decreasing = TRUE)
keytypes(org.Hs.eg.db)
head(ROSI_lfc_D10)

#gene set enrichment
gse_ROSI_lfc_D10 <- gseGO(geneList=ROSI_lfc_D10, 
             ont = "All", 
             keyType = "SYMBOL", 
             minGSSize = 3, 
             maxGSSize = 300, 
             pvalueCutoff = 0.05,
             nPermSimple= 10000,
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")
gse_ROSI_lfc_D10
dotplot(gse_ROSI_lfc_D10,showCategory=20) + ggtitle("GSEA ROSI vs DMSO D10")
#### Reduce clusters######
library(rrvgo)
simMatrix_ROSI_D10 <- calculateSimMatrix(gse_ROSI_lfc_D10@result$ID, orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")

scores_simMatrix_ROSI_D10 <- setNames(-log10(gse_ROSI_lfc_D10@result$qvalues), gse_ROSI_lfc_D10@result$ID)
reducedTerms_ROSI_D10 <- reduceSimMatrix(simMatrix_ROSI_D10, scores_simMatrix_ROSI_D10, threshold = 0.7, orgdb = "org.Hs.eg.db")
#heatmapPlot(simMatrix_ROSI_D10,reducedTerms_ROSI_D10,annotateParent = T,annotationLabel = "parentTerm",fontsize= 6) # this is super noisy
scatterPlot(simMatrix_ROSI_D10,reducedTerms_ROSI_D10)
treemapPlot(reducedTerms_ROSI_D10)
###UP regulated GSEA###
ROSI_lfc_D10_up <- ROSI_lfc_D10[ROSI_lfc_D10>0]
length(ROSI_lfc_D10_up)#108 up regulated genes
gse_ROSI_lfc_D10_Up <- gseGO(geneList=ROSI_lfc_D10_up, 
             ont = "BP", 
             keyType = "SYMBOL", 
             minGSSize = 3, 
             maxGSSize = 300, 
             pvalueCutoff = 0.05,
             nPermSimple= 10000,
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")
#write.csv(gse_ROSI_lfc_D10_Up,"~/Documents/Git/MSC_Bulk/RNA_seq_Data/gse_ROSI_lfc_D10_Up.csv", quote = F)
dotplot(gse_ROSI_lfc_D10_Up,showCategory=10) + ggtitle("GSEA ROSI vs DMSO D10 Upregulated")

### Seperating up vs down regulated genes ####
####Up Regulated DEGS####
res_ROSI_contrD10_anno_padj_0.05 <- res_ROSI_contrD10_anno_padj_0.05[order(res_ROSI_contrD10_anno_padj_0.05$padj),]
dim(res_ROSI_contrD10_anno_padj_0.05)#973 7 
head(res_ROSI_contrD10_padj_0.05_NC)
dim(res_ROSI_contrD10_padj_0.05_NC) #973 6 
#normalized counts upregulated
res_ROSI_contrD10_padj_0.05_NC_UP <- res_ROSI_contrD10_padj_0.05_NC[rownames(res_ROSI_contrD10_anno_padj_0.05[which(res_ROSI_contrD10_anno_padj_0.05$log2FoldChange>0),]),] #600 6
dim(res_ROSI_contrD10_padj_0.05_NC_UP) #600 6
top_ROSI_D10_Up <- data.frame(res_ROSI_contrD10_padj_0.05_NC_UP[1:50,]) %>% rownames_to_column(var ="ensgene")
dim(top_ROSI_D10_Up)
head(top_ROSI_D10_Up)
top_ROSI_D10_Up <- gather(top_ROSI_D10_Up, key ="samplename", value ="normalized_counts",2:7)
head(top_ROSI_D10_Up)

top_ROSI_D10_Up<- inner_join(top_ROSI_D10_Up,
                     rownames_to_column(coldata_ROSI, var ="samplename"),
                     by ="samplename")
head(top_ROSI_D10_Up)
top_ROSI_up_D10_anno <- data.frame(top_ROSI_D10_Up)
top_ROSI_up_D10_anno$symbol<-                                                                                      mapIds(edb,keys=top_ROSI_D10_Up$ensgene,column="SYMBOL",keytype="GENEID",multiVals="first")
top_ROSI_up_D10_anno$symbol

ggplot(top_ROSI_up_D10_anno)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top Upregulated Genes ROSI vs DMSO Day 3")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))
  
######Removing RP genes that may be altering the data#####
head(res_ROSI_contrD10_anno_padj_0.05)
res_ROSI_contrD10_anno_padj_0.05_noRP11 <- res_ROSI_contrD10_anno_padj_0.05[which(substr(res_ROSI_contrD10_anno_padj_0.05$symbol,1,4) !="RP11"),]
dim(res_ROSI_contrD10_anno_padj_0.05_noRP11)# Removed 167 genes  806 7
library(writexl)
#writexl::write_xlsx(res_ROSI_contrD10_anno_padj_0.05_noRP11,"~/Documents/Git/MSC_Bulk/RNA_seq_Data/ROSI_noMSC_DEGs_D10_sig_anno_nnoRP11genes.xlsx")

####Down regulated DEGs####
res_ROSI_contrD10_anno_padj_0.05 <- res_ROSI_contrD10_anno_padj_0.05[order(res_ROSI_contrD10_anno_padj_0.05$padj),]
head(res_ROSI_contrD10_padj_0.05_NC)
#normalized counts upregulated
res_ROSI_contrD10_padj_0.05_NC_Down <- res_ROSI_contrD10_padj_0.05_NC[rownames(res_ROSI_contrD10_anno_padj_0.05[which(res_ROSI_contrD10_anno_padj_0.05$log2FoldChange<0),]),]
dim(res_ROSI_contrD10_padj_0.05_NC_Down) #373 6
#dim(res_sig_normalized_counts_ROSI_Down) #156 27  

top_ROSI_D10_Down <- data.frame(res_ROSI_contrD10_padj_0.05_NC_Down[1:50,]) %>% rownames_to_column(var ="ensgene")
dim(top_ROSI_D10_Down)#20 28
head(top_ROSI_D10_Down)
top_ROSI_D10_Down <- gather(top_ROSI_D10_Down, key ="samplename", value ="normalized_counts",2:7)
head(top_ROSI_D10_Down)

top_ROSI_D10_Down<- inner_join(top_ROSI_D10_Down,
                     rownames_to_column(coldata_ROSI, var ="samplename"),
                     by ="samplename")
head(top_ROSI_D10_Down)
top_ROSI_D10_Down_anno <- data.frame(top_ROSI_D10_Down)
dim(top_ROSI_D10_Down)
top_ROSI_D10_Down_anno$symbol<-                                                                                      mapIds(edb,keys=top_ROSI_D10_Down$ensgene,column="SYMBOL",keytype="GENEID",multiVals="first")
top_ROSI_D10_Down_anno$symbol

ggplot(top_ROSI_D10_Down_anno)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top Downregulated Genes ROSI vs DMSO Day 3")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))
```
###D14
```{r Day 14 comparison}
#BiocManager::install("DESeq2")
suppressPackageStartupMessages(library("DESeq2"))

#Subsetting the data to only have the pertinent information
# cts_mat_ROSI<- cts_mat[,which(substr(colnames(cts_mat),2,2)=="R"|substr(colnames(cts_mat),2,2)=="D")]
# rownames(cts_mat_ROSI) <- substr(rownames(cts_mat_ROSI),1,15)#Removing the .# on the gene names
# coldata_ROSI <- coldata_norm[which(substr(rownames(coldata_norm),2,2)=="R"|substr(rownames(coldata_norm),2,2)=="D"),]
colnames(cts_mat_ROSI) #58037 27
# dim(coldata_ROSI) #27 4
cts_mat_ROSI_D14 <- cts_mat_ROSI[,22:27]
coldata_ROSI_D14 <- coldata_ROSI[22:27,]
##Creating DDS object from the subset data                                  
dds_ROSI_D14 <- DESeqDataSetFromMatrix(countData = cts_mat_ROSI_D14,
                                                 colData = coldata_ROSI_D14,
                                                 design = ~ condition                                               )
summary(dds_ROSI_D14)
head(dds_ROSI_D14)   
###Adding meta data columns
rownames(dds_ROSI_D14) <- substr(rownames(dds_ROSI_D14),1,15)
featureData <- data.frame(gene=substr(rownames(cts_mat_ROSI_D14),1,15))
mcols(dds_ROSI_D14) <- DataFrame(mcols(dds_ROSI_D14), featureData)
mcols(dds_ROSI_D14)
featureData$symbol <- cts_1021$SYMBOL# Adding mcol symbol data
mcols(dds_ROSI_D14) <- DataFrame(mcols(dds_ROSI_D14), featureData)
mcols(dds_ROSI_D14)
###Removing low quality rows
keep <- rowSums(counts(dds_ROSI_D14)) >= 10 #30255 genes passed this low bar
dds_ROSI_D14 <- dds_ROSI_D14[keep,]
####Making things factors
dds_ROSI_D14$condition <- factor(dds_ROSI_D14$condition, levels = c("control","ROSI"))
dds_ROSI_D14$condition
# Determine the size factors to use for normalization
dds_ROSI_D14<- estimateSizeFactors(dds_ROSI_D14)
sizeFactors(dds_ROSI_D14)
# Extract the normalized counts
normalized_counts_ROSI_D14 <- counts(dds_ROSI_D14, normalized=T)
head(normalized_counts_ROSI_D14)

# Transform the normalized counts 
vsd_ROSI_D14 <- vst(dds_ROSI_D14,blind=F )

# Extract the matrix of transformed counts
vsd_mat_ROSI_D14 <- assay(dds_ROSI_D14)

# Compute the correlation values between samples
vsd_cor_ROSI_D14 <- cor(vsd_mat_ROSI_D14) 

# Plot the heatmap
pheatmap(vsd_cor_ROSI_D14)

colData(vsd_ROSI_D14)
###Plot PCAs####
plotPCA(vsd_ROSI_D14, intgroup= "condition")

plotPCA(vsd_ROSI_D14, intgroup = c("Time","condition"))

DES_ROSI_D14 <- DESeq(dds_ROSI_D14)
res_ROSI_D14 <- results(DES_ROSI_D14)
resultsNames(res_ROSI_D14)
head(res_ROSI_D14)
mcols(res_ROSI_D14)
#Increase Alpha threshold
res_ROSI_D14_contrast <- results(DES_ROSI_D14, contrast=c("condition","ROSI","control"),alpha = 0.05)
head(res_ROSI_D14_contrast)

####DEG L2fc threshold alteration########
# #BiocManager::install("apeglm")
# #library(apeglm)
# # Shrink the log2 fold change estimates to be more accurate
# res_contr_ROSI_lfc_D14 <- results(DES_ROSI_D14, 
#                         contrast=c("condition","ROSI","control"),
#                         alpha = 0.05,
#                         lfcThreshold = 0.2)
# resultsNames(dds_ROSI_D14)
# 
# resLFC_ROSI_D14 <- lfcShrink(dds_ROSI_D14,
#                     contrast = c("condition","ROSI","control"),
#                     res = res_contr_ROSI_lfc_D14)
# summary(res_contr_ROSI_lfc_D14)
# #Save results as a dataframe
# res_cont_ROSI_D14_lfc.2 <- data.frame(res_contr_ROSI_lfc_D14)
#res_Ordered <- res_cont_ROSI[order(res_cont_ROSI$pvalue),]
#summary(res_Ordered)

### Subsetting for padj and log2fc data ####
sum(res_ROSI_D14_contrast$padj < 0.05, na.rm=TRUE) #69 lowering the Padjusted value removed some of the genes
res_ROSI_D14_contrast_padj_0.05 <- res_ROSI_D14_contrast[which(res_ROSI_D14_contrast$padj<0.05),]


#All significant normalized counts table
res_order_ROSI_D14 <-res_ROSI_D14_contrast_padj_0.05[order(res_ROSI_D14_contrast_padj_0.05$padj),]

res_sig_order_ROSIcontrD14_padj_0.05 <- res_ROSI_D14_contrast_padj_0.05[order(res_ROSI_D14_contrast_padj_0.05$padj),]
dim(res_order_ROSI_D14)#382 6


##Annnotate significant table
suppressPackageStartupMessages(library(org.Hs.eg.db))
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(EnsDb.Hsapiens.v86)
edb <- EnsDb.Hsapiens.v86
organism(edb)
keytypes(edb)
keys(edb,keytype = "GENEID")
###Annotate results table#####
res_ROSI_contrD14_anno <- data.frame(res_order_ROSI_D14)
res_ROSI_contrD14_anno$symbol <- mapIds(edb,keys=rownames(res_ROSI_contrD14_anno),column="SYMBOL",keytype="GENEID",multiVals="first")

library(writexl)
#writexl::write_xlsx(res_ROSI_contrD14_anno,"~/Documents/Git/MSC_Bulk/RNA_seq_Data/ROSI_noMSC_DEGs_D14_sig_anno.xlsx")
```
####D14 Visualization
```{r D14 samples only results}
# trying to run contrast on the contrasted data

head(res_ROSI_contrD14_anno)

## Subsetting significant counts
sum(res_ROSI_contrD14_anno$padj < 0.05, na.rm=TRUE) #21 A lot more significant genes from day 7 to day 14
res_ROSI_contrD14_anno_padj_0.05 <- res_ROSI_contrD14_anno[which(res_ROSI_contrD14_anno$padj<0.05),]
dim(res_ROSI_contrD14_anno_padj_0.05)#69 7
###l2fc cutoff
res_ROSI_contrD14_anno_padj_0.05_l2fc1 <- res_ROSI_contrD14_anno_padj_0.05[which(abs(res_ROSI_contrD14_anno_padj_0.05$log2FoldChange)>1),]
dim(res_ROSI_contrD14_anno_padj_0.05_l2fc1)#19 7
#Data to use: 
#           res_ROSI_anno(all results annotated with threshold cutoff)
#normalized counts tables:
#           vsd_mat_ROSI(this is the normalized data)
#           res_sig_normalized_counts_ROSI(normalized data subset form the standard constrast)

res_ROSI_contrD14_padj_0.05_NC <- vsd_mat_ROSI_D14[rownames(res_ROSI_contrD14_anno_padj_0.05),]
res_ROSI_contrD14_anno_padj_0.05_l2fc_NC <- vsd_mat_ROSI_D14[rownames(res_ROSI_contrD14_anno_padj_0.05_l2fc1),]
####Make the normalized counts from day contrast#####
head(res_ROSI_contrD14_anno_padj_0.05)
head(res_ROSI_contrD14_padj_0.05_NC)
##Heatmap####
library(pheatmap)
library(ComplexHeatmap)


pheatmap(res_ROSI_contrD14_padj_0.05_NC,
         cluster_rows =T,
         color = heat_colors,
         cluster_cols = F,
         show_rownames =F,
         annotation = select(coldata_ROSI, condition, Day),
         scale = "row")
colnames(res_sig_normalized_counts_ROSI)

####Volcano#######
##Annnotate significant table
suppressPackageStartupMessages(library(org.Hs.eg.db))
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(EnsDb.Hsapiens.v86)
edb <- EnsDb.Hsapiens.v86
organism(edb)
keytypes(edb)
keys(edb,keytype = "GENEID")
###Annotate results table#####
res_ROSI_D14_contrast_anno <- data.frame(res_ROSI_D14_contrast)
res_ROSI_D14_contrast_anno$symbol <- mapIds(edb,keys=rownames(res_ROSI_D14_contrast),column="SYMBOL",keytype="GENEID",multiVals="first")
### Volcano plots######

EnhancedVolcano(res_ROSI_D14_contrast_anno,
    lab = res_ROSI_D14_contrast_anno$symbol,
    x = 'log2FoldChange',
    y = 'padj',
    pCutoff = 5e-2,
    FCcutoff = .75,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 6.0,
    title = "results ROSI vs DMSO contrast day 14",
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 4.0,
    drawConnectors = T,
    widthConnectors = 0.75,
    max.overlaps = 50)
###GO analysis####
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(ggplot2)
# we use ggplot2 to add x axis labels (ex: ridgeplot)
head(res_ROSI_contrD14_anno_padj_0.05)
ROSI_lfc_D14 <- res_ROSI_contrD14_anno_padj_0.05$log2FoldChange
names(ROSI_lfc_D14) <- res_ROSI_contrD14_anno_padj_0.05$symbol
ROSI_lfc_D14<-na.omit(ROSI_lfc_D14)
# sort the list in decreasing order (required for clusterProfiler)
ROSI_lfc_D14 = sort(ROSI_lfc_D14, decreasing = TRUE)
keytypes(org.Hs.eg.db)
head(ROSI_lfc_D14)

#gene set enrichment
gse_ROSI_lfc_D14 <- gseGO(geneList=ROSI_lfc_D14, 
             ont = "All", 
             keyType = "SYMBOL", 
             minGSSize = 3, 
             maxGSSize = 300, 
             pvalueCutoff = 0.05,
             nPermSimple= 10000,
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")
gse_ROSI_lfc_D14
dotplot(gse_ROSI_lfc_D14,showCategory=10) + ggtitle("GSEA ROSI vs DMSO D14")
#### Reduce clusters######
library(rrvgo)
simMatrix_ROSI_D14 <- calculateSimMatrix(gse_ROSI_lfc_D14@result$ID, orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")

scores_simMatrix_ROSI_D14 <- setNames(-log10(gse_ROSI_lfc_D14@result$qvalues), gse_ROSI_lfc_D14@result$ID)
reducedTerms_ROSI_D14 <- reduceSimMatrix(simMatrix_ROSI_D14, scores_simMatrix_ROSI_D14, threshold = 0.7, orgdb = "org.Hs.eg.db")
#heatmapPlot(simMatrix_ROSI_D14,reducedTerms_ROSI_D14,annotateParent = T,annotationLabel = "parentTerm",fontsize= 6) # this is super noisy
scatterPlot(simMatrix_ROSI_D14,reducedTerms_ROSI_D14)
treemapPlot(reducedTerms_ROSI_D14)

###UP regulated GSEA###
ROSI_lfc_D14_up <- ROSI_lfc_D14[ROSI_lfc_D14>0]
length(ROSI_lfc_D14_up)#27 up regulated genes
gse_ROSI_lfc_D14_Up <- gseGO(geneList=ROSI_lfc_D14_up, 
             ont = "BP", 
             keyType = "SYMBOL", 
             minGSSize = 3, 
             maxGSSize = 300, 
             pvalueCutoff = 0.05,
             nPermSimple= 10000,
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")
#write.csv(gse_ROSI_lfc_D14_Up,"~/Documents/Git/MSC_Bulk/RNA_seq_Data/gse_ROSI_lfc_D14_Up.csv", quote = F)
dotplot(gse_ROSI_lfc_D14_Up,showCategory=10) + ggtitle("GSEA ROSI vs DMSO D14 Upregulated")

### Seperating up vs down regulated genes ####
####Up Regulated DEGS####
res_ROSI_contrD14_anno_padj_0.05 <- res_ROSI_contrD14_anno_padj_0.05[order(res_ROSI_contrD14_anno_padj_0.05$padj),]
head(res_ROSI_contrD14_padj_0.05_NC)
#normalized counts upregulated
res_ROSI_contrD14_padj_0.05_NC_UP <- res_ROSI_contrD14_padj_0.05_NC[rownames(res_ROSI_contrD14_anno_padj_0.05[which(res_ROSI_contrD14_anno_padj_0.05$log2FoldChange>0),]),]
dim(res_ROSI_contrD14_padj_0.05_NC_UP) #27 6
top_ROSI_D14_Up <- data.frame(res_ROSI_contrD14_padj_0.05_NC_UP) %>% rownames_to_column(var ="ensgene")
dim(top_ROSI_D14_Up)
head(top_ROSI_D14_Up)
top_ROSI_D14_Up <- gather(top_ROSI_D14_Up, key ="samplename", value ="normalized_counts",2:7)
head(top_ROSI_D14_Up)

top_ROSI_D14_Up<- inner_join(top_ROSI_D14_Up,
                     rownames_to_column(coldata_ROSI, var ="samplename"),
                     by ="samplename")
head(top_ROSI_D14_Up)
top_ROSI_up_D14_anno <- data.frame(top_ROSI_D14_Up)
top_ROSI_up_D14_anno$symbol<-                                                                                      mapIds(edb,keys=top_ROSI_D14_Up$ensgene,column="SYMBOL",keytype="GENEID",multiVals="first")
top_ROSI_up_D14_anno$symbol

ggplot(top_ROSI_up_D14_anno)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top Upregulated Genes ROSI vs DMSO Day 3")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))
        
####Down regulated DEGs####
res_ROSI_contrD14_anno_padj_0.05 <- res_ROSI_contrD14_anno_padj_0.05[order(res_ROSI_contrD14_anno_padj_0.05$padj),]
head(res_ROSI_contrD14_padj_0.05_NC)
#normalized counts upregulated
res_ROSI_contrD14_padj_0.05_NC_Down <- res_ROSI_contrD14_padj_0.05_NC[rownames(res_ROSI_contrD14_anno_padj_0.05[which(res_ROSI_contrD14_anno_padj_0.05$log2FoldChange<0),]),]
dim(res_ROSI_contrD14_padj_0.05_NC_Down) #42 6
#dim(res_sig_normalized_counts_ROSI_Down) #156 27  

top_ROSI_D14_Down <- data.frame(res_ROSI_contrD7_anno_padj_0.05_NC_Down[1:50,]) %>% rownames_to_column(var ="ensgene")
dim(top_ROSI_D14_Down)#20 28
head(top_ROSI_D14_Down)
top_ROSI_D14_Down <- gather(top_ROSI_D14_Down, key ="samplename", value ="normalized_counts",2:28)
head(top_ROSI_D14_Down)

top_ROSI_D14_Down<- inner_join(top_ROSI_D14_Down,
                     rownames_to_column(coldata_ROSI, var ="samplename"),
                     by ="samplename")
head(top_ROSI_D14_Down)
top_ROSI_Down_anno <- data.frame(top_ROSI_D14_Down)
dim(top_ROSI_D14_Down)
top_ROSI_D14_Down_anno$symbol<-                                                                                      mapIds(edb,keys=top_ROSI_D14_Down$ensgene,column="SYMBOL",keytype="GENEID",multiVals="first")
top_ROSI_D14_Down_anno$symbol

ggplot(top_ROSI_D14_Down_anno)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top Downregulated Genes ROSI vs DMSO Day 3")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))

```
##D14 vs D14 DEseq object AT check
```{r D14 vs D14 adipo check}
head(res_SD_NoExtra_alph05_anno)
vsd_mat_ROSI_D14
vsd_mat_ROSI_D14_anno <- as.data.frame(vsd_mat_ROSI_D14) %>% rownames_to_column(var ="ensgene")
vsd_mat_ROSI_D14_anno <- left_join(x = vsd_mat_ROSI_D14_anno,
             y = grch38[,c("ensgene","symbol","description")],
             by ="ensgene")
WG_ROSI_D14 <-grep(vsd_mat_ROSI_D14_anno$symbol, pattern = paste(white_genes,collapse = "|"))
WG_ROSI_D14_anno <- vsd_mat_ROSI_D14_anno[WG_ROSI_D14,]
length(white_genes)

Beige_genes <- c("AQP7","ASC1","CAR4","CD137","CITED1","EAR2","NR2F6","SHOX2","SLC27A1","SP100","TBX1","TMEM26","CIDEA","COX7A","EPSTI1","FGF21","HSPB7","KCNK3","LHX8","KLF11")
BeG_ROSI_D14 <-grep(vsd_mat_ROSI_D14_anno$symbol, pattern = paste(Beige_genes,collapse = "|"))
BeG_ROSI_D14_anno <- vsd_mat_ROSI_D14_anno[BeG_ROSI_D14,]
BeG_ROSI_D14_anno <- unique(BeG_ROSI_D14_anno)
length(Beige_genes)
Brown_genes <- c("BMP7","EBF2","EDNRB","EVA1","MIR133B","MIR206","MYF5","PDK4","PREX1","ZIC1","UCP1","PPARA","CIDEA")
BrG_ROSI_D14 <-grep(vsd_mat_ROSI_D14_anno$symbol, pattern = paste(Brown_genes,collapse = "|"))
BrG_ROSI_D14_anno <- vsd_mat_ROSI_D14_anno[BrG_ROSI_D14,]
length(Brown_genes)
#BrG_ROSI_D14_anno
```


```{r D14 vs D14 visualizing Adpocyte types}
#Getting count data for these genes


rownames(WG_ROSI_D14_anno) <- WG_ROSI_D14_anno$symbol
res_ROSI_D14_normalized_counts_WAT_exactmatch <- WG_ROSI_D14_anno[which(rownames(WG_ROSI_D14_anno) %in% white_genes),]

dim(res_ROSI_D14_normalized_counts_WAT_exactmatch) #18 9
#Heat map of Gene panels
pheatmap(res_ROSI_D14_normalized_counts_WAT_exactmatch[2:7],
         color = heat_colors,
         cluster_rows =T,
         show_rownames =T,
         annotation = select(coldata_ROSI_D14, condition),
         scale ="row")

#Getting count data for these genes
rownames(BeG_ROSI_D14_anno) <- BeG_ROSI_D14_anno$symbol
#subsetting for exact gene names
res_ROSI_D14_normalized_counts_BeAT_exactmatch <- BeG_ROSI_D14_anno[(rownames(BeG_ROSI_D14_anno) %in% Beige_genes),]

#Heat map of Gene panels
dim(res_ROSI_D14_normalized_counts_BeAT_exactmatch)
pheatmap(res_ROSI_D14_normalized_counts_BeAT_exactmatch[2,7],
         color = heat_colors,
         cluster_rows =T,
         show_rownames =T,
         annotation = select(coldata_ROSI_D14_Noextra, Time),
         scale ="row")

#Getting count data for these genes

rownames(BrG_ROSI_D14_anno) <- BrG_ROSI_D14_anno$symbol
#Subsetting for Exact gene name matches
res_ROSI_D14_normalized_counts_BAT_exactmatch <- BrG_ROSI_D14_anno[which(rownames(BrG_ROSI_D14_anno) %in% Brown_genes),]
#Heat map of Gene panels
dim(res_ROSI_D14_normalized_counts_BAT_exactmatch)
pheatmap(res_ROSI_D14_normalized_counts_BAT_exactmatch[2,7],
         color = heat_colors,
         cluster_rows =T,
         show_rownames =T,
         annotation = select(coldata_ROSI_D14_Noextra, Time),
         scale ="row")

```

### diuvergent bar pot for adipogenic differentiaiton
```{r Divergen Bar plot Adipocyte type}
#Normalizing the data
res_ROSI_D14_normalized_counts_AT_d14 <- res_ROSI_D14_normalized_counts_AT[,which(substr(colnames(res_ROSI_D14_normalized_counts_AT),1,1)=="E")]
SD_D14_avg_adipogenes_ROSI_D14 <- data.frame(SD_D14=rowMeans(res_ROSI_D14_normalized_counts_AT_d14[,1:3]))
SD_D14_avg_adipogenes_ROSI_D14 <- data.frame(log2(SD_D14_avg_adipogenes_ROSI_D14))
res_ROSI_D14_normalized_counts_AT_d14_normed <- log2(res_ROSI_D14_normalized_counts_AT_d14[,1:6])-SD_D14_avg_adipogenes_ROSI_D14$SD_D14
res_ROSI_D14_normalized_counts_AT_d14_normed

#####Divergent bar graph ############
library(matrixStats)
#Making data frame of means 
df_AT_ROSI_D14_log2_normed <- data.frame(Day_14_DMSO=rowMeans(res_ROSI_D14_normalized_counts_AT_d14_normed[,1:3]),Day_14_ROSI=rowMeans(res_ROSI_D14_normalized_counts_AT_d14_normed[,4:6]),row.names = rownames(res_ROSI_D14_normalized_counts_AT_d14_normed),gene = rownames(res_ROSI_D14_normalized_counts_AT_d14_normed))
head(df_AT_ROSI_D14_log2_normed)

# lock in factor level order
df_AT_ROSI_D14_log2_normed$gene <- factor(df_AT_ROSI_D14_log2_normed$gene, levels = df_AT_ROSI_D14_log2_normed$gene)
#inverting the order because the data keeps plotting upside down
df_AT_ROSI_D14_log2_normed_reorder=df_AT_ROSI_D14_log2_normed[order(nrow(df_AT_ROSI_D14_log2_normed):1),]
df_AT_ROSI_D14_log2_normed_reorder
df_AT_ROSI_D14_log2_normed_reorder$gene <- factor(df_AT_SD_NE_log2_normed_reorder$gene, levels = df_AT_ROSI_D14_log2_normed_reorder$gene)
library(ggplot2)
###Colors
library(MetBrewer)
degascolor <- met.brewer(name = 'Morgenstern',7,type = "discrete")
###Control Adipocytes
color <- ifelse(df_AT_ROSI_D14_log2_normed_reorder$Day_14_ROSI < 0, degascolor[6], degascolor[4])
ggplot(df_AT_ROSI_D14_log2_normed_reorder, aes(y = gene, x = Day_14_ROSI)) +
  geom_bar(stat = "identity",
           show.legend = FALSE, fill = color) + # Remove the legend +
  xlab("ROSI log2 Relative Gene expression to D14 Standard adipocyte") +
  ylab("Day_14_DMSO")
```

##Top DEGs from all the analysis
I am collecting all the genes from the noMSC analysis, and day by day analysis to crearte a master list of genes that are DEGs and then tusing those to see a time course of the top DEGs over time. 
I will mostly focus on the overlap of the complete noMSc analysis and the DAy 10/14 comparisons because those are the most likely to be showing the ROSI effects. 
```{r DEG aggregation}
###NoMSC Analysis####
#All significant normalized counts table
res_order_ROSI <-res_contr_ROSI[order(res_contr_ROSI$padj),]
#res_sig_order_ROSI_padj_0.1 <-res_contr_ROSI_padj_0.1[order(res_contr_ROSI_padj_0.1$padj),]
#res_sig_order_ROSI_padj_0.05 <- res_contr_ROSI_padj_0.05[order(res_contr_ROSI_padj_0.05$padj),]
dim(res_sig_order_ROSI)#210 6
#res_sig_normalized_counts_ROSI <- vsd_mat_ROSI[substr(rownames(res_sig_order_ROSI_padj_0.05),1,15),] 
dim(res_sig_normalized_counts_ROSI) #118 27
####Significant up and down for noExtra NoMSC comparison###
#res_ROSI_sig_down <-res_sig_order_ROSI_padj_0.05[which(res_sig_order_ROSI_padj_0.05$log2FoldChange<0),]
dim(res_ROSI_sig_down) #93 6
#res_ROSI_sig_up <-res_sig_order_ROSI_padj_0.05[which(res_sig_order_ROSI_padj_0.05$log2FoldChange>0),]
dim(res_ROSI_sig_up) #25 6
#Normalized counts table for up and down regulations
# res_sig_normalized_counts_ROSI_up <- vsd_mat_ROSI[rownames(res_ROSI_sig_up[order(res_ROSI_sig_up$padj),]),]
# res_sig_normalized_counts_ROSI_Down <- vsd_mat_ROSI[rownames(res_ROSI_sig_down[order(res_ROSI_sig_down$padj),]),] 
str(res_sig_normalized_counts_ROSI_up)#25 27
str(res_sig_normalized_counts_ROSI_Down)#93 27
##Annnotate significant table
suppressPackageStartupMessages(library(org.Hs.eg.db))
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(EnsDb.Hsapiens.v86)
edb <- EnsDb.Hsapiens.v86
organism(edb)
keytypes(edb)
keys(edb,keytype = "GENEID")
###Annotate results table#####
# res_ROSI_noMSC_anno <- data.frame(res_order_ROSI)
# res_ROSI_noMSC_anno$symbol <- mapIds(edb,keys=rownames(res_order_ROSI),column="SYMBOL",keytype="GENEID",multiVals="first")
####DEGs####
# ROSI_DEGS_anno <- data.frame(res_sig_order_ROSI)
# ROSI_DEGS_anno$symbol<-                                                                                      mapIds(edb,keys=rownames(res_sig_order_ROSI),column="SYMBOL",keytype="GENEID",multiVals="first")
# ROSI_DEGS_anno$symbol
# ROSI_DEGS_anno_padj0.05 <- ROSI_DEGS_anno[which(rownames(ROSI_DEGS_anno) %in% rownames(res_sig_order_ROSI_padj_0.05)),]
str(ROSI_DEGS_anno_padj0.05)##118
upreg_ROSI_DEGS_anno_padj0.05 <- ROSI_DEGS_anno_padj0.05[which(ROSI_DEGS_anno_padj0.05$log2FoldChange>0),]
dim(upreg_ROSI_DEGS_anno_padj0.05) #25 7
####Day 10 #####
####Up Regulated DEGS####
#res_ROSI_contrD10_anno_padj_0.05 <- res_ROSI_contrD10_anno_padj_0.05[order(res_ROSI_contrD10_anno_padj_0.05$padj),]
dim(res_ROSI_contrD10_anno_padj_0.05)#973 7 
head(res_ROSI_contrD10_padj_0.05_NC)
dim(res_ROSI_contrD10_padj_0.05_NC) #973 6 
#normalized counts upregulated
# res_ROSI_contrD10_padj_0.05_NC_UP <- res_ROSI_contrD10_padj_0.05_NC[rownames(res_ROSI_contrD10_anno_padj_0.05[which(res_ROSI_contrD10_anno_padj_0.05$log2FoldChange>0),]),] #600 6
dim(res_ROSI_contrD10_padj_0.05_NC_UP) #600 6
####Down regulated DEGs####
#res_ROSI_contrD10_anno_padj_0.05 <- res_ROSI_contrD10_anno_padj_0.05[order(res_ROSI_contrD10_anno_padj_0.05$padj),]
head(res_ROSI_contrD10_padj_0.05_NC)
#normalized counts upregulated
# res_ROSI_contrD10_padj_0.05_NC_Down <- res_ROSI_contrD10_padj_0.05_NC[rownames(res_ROSI_contrD10_anno_padj_0.05[which(res_ROSI_contrD10_anno_padj_0.05$log2FoldChange<0),]),]
dim(res_ROSI_contrD10_anno_padj_0.05_NC_Down) #108 7
#dim(res_sig_normalized_counts_ROSI_Down) #156 27  

####Day14###
head(res_ROSI_contrD14_anno)# Day 14 vs 14 results

## Subsetting significant counts
# sum(res_ROSI_contrD14_anno$padj < 0.05, na.rm=TRUE) #21 A lot more significant genes from day 7 to day 14
 res_ROSI_contrD14_anno_padj_0.05 <- res_ROSI_contrD14_anno[which(res_ROSI_contrD14_anno$padj<0.05),]
dim(res_ROSI_contrD14_anno_padj_0.05)#69 7
###l2fc cutoff
# res_ROSI_contrD14_anno_padj_0.05_l2fc1 <- res_ROSI_contrD14_anno_padj_0.05[which(abs(res_ROSI_contrD14_anno_padj_0.05$log2FoldChange)>1),]
dim(res_ROSI_contrD14_anno_padj_0.05_l2fc1)#19 7
#Data to use: 
#           res_ROSI_anno(all results annotated with threshold cutoff)
#normalized counts tables:
#           vsd_mat_ROSI(this is the normalized data)
#           res_sig_normalized_counts_ROSI(normalized data subset form the standard constrast)
# res_ROSI_contrD14_padj_0.05_NC <- vsd_mat_ROSI_D14[rownames(res_ROSI_contrD14_anno_padj_0.05),]
# res_ROSI_contrD14_anno_padj_0.05_l2fc_NC <- vsd_mat_ROSI_D14[rownames(res_ROSI_contrD14_anno_padj_0.05_l2fc1),]
####Make the normalized counts from day contrast#####
head(res_ROSI_contrD14_anno_padj_0.05)
head(res_ROSI_contrD14_padj_0.05_NC)
#normalized counts upregulated
dim(res_ROSI_contrD14_padj_0.05_NC_UP) #27 6
dim(res_ROSI_contrD14_padj_0.05_NC_Down)# 42 6
###Start Combining genes here####
dim(ROSI_DEGS_anno_padj0.05) #No MSC all samples results 118 genes
head(res_sig_normalized_counts_ROSI)#Normalized counts for No MSC all sample results
##D14 results
dim(res_ROSI_contrD14_anno_padj_0.05)#69 genes
head(res_ROSI_contrD14_padj_0.05_NC)
##D10 results
dim(res_ROSI_contrD10_anno_padj_0.05)#973 genes
head(res_ROSI_contrD10_padj_0.05_NC)
###Master Table
res_ROSI_anno_noMSC_D10_D14 <- rbind(ROSI_DEGS_anno_padj0.05,res_ROSI_contrD10_anno_padj_0.05,res_ROSI_contrD14_anno_padj_0.05)
dim(res_ROSI_anno_noMSC_D10_D14)
library(writexl)
#write_xlsx(res_ROSI_anno_noMSC_D10_D14,path = "~/Documents/Git/MSC_Bulk/RNA_seq_Data/res_ROSI_anno_noMSC_D10_D14_master.xlsx")
res_ROSI_anno_noMSC_D10_D14_duplicated <- res_ROSI_anno_noMSC_D10_D14[which(duplicated(res_ROSI_anno_noMSC_D10_D14$symbol) |duplicated(res_ROSI_anno_noMSC_D10_D14$symbol, fromLast = T)),]

res_ROSI_anno_noMSC_D10_D14_duplicated <- res_ROSI_anno_noMSC_D10_D14_duplicated[order(res_ROSI_anno_noMSC_D10_D14_duplicated$symbol),]
#write_xlsx(res_ROSI_anno_noMSC_D10_D14_duplicated,path = "~/Documents/Git/MSC_Bulk/RNA_seq_Data/res_ROSI_anno_noMSC_D10_D14_duplicated.xlsx")

###up Down###
res_ROSI_anno_noMSC_D10_D14_duplicated_up <- res_ROSI_anno_noMSC_D10_D14_duplicated[which(res_ROSI_anno_noMSC_D10_D14_duplicated$log2FoldChange>0),]

res_ROSI_anno_noMSC_D10_D14_duplicated_down <- res_ROSI_anno_noMSC_D10_D14_duplicated[which(res_ROSI_anno_noMSC_D10_D14_duplicated$log2FoldChange<0),]

library(annotables)
#annotate the genes
##Normalized counts for aggreagate genes
dim(res_ROSI_anno_noMSC_D10_D14) #1160 7 
dim(vsd_mat_ROSI)
res_sig_normalized_counts_ROSI_aggregate <- vsd_mat_ROSI[which(rownames(vsd_mat_ROSI) %in% rownames(res_ROSI_anno_noMSC_D10_D14)),]

length(which(duplicated(res_ROSI_anno_noMSC_D10_D14$symbol)))

##the missing rows are the duplicated rows they have the slightly altered rownames they are all 16 characters long 
# missing_rownames <- rownames(res_ROSI_anno_noMSC_D10_D14)[which(!(rownames(res_ROSI_anno_noMSC_D10_D14) %in% rownames(vsd_mat_ROSI)))]
# which(substr(rownames(res_ROSI_anno_noMSC_D10_D14),16,16) ==1)
# which(substr(rownames(res_ROSI_anno_noMSC_D10_D14),16,16) ==2)
# which(rownames(vsd_mat_ROSI)%in%missing_rownames)

str(res_sig_normalized_counts_ROSI_aggregate)#1088 27 this probably removed the duplicated genes and only called them once
####Top gene graph####
top_ROSI_aggregate <- data.frame(res_ROSI_anno_noMSC_D10_D14) %>% rownames_to_column(var ="ensgene")
dim(top_ROSI_aggregate)
top_ROSI_aggregate <- gather(top_ROSI_aggregate, key ="samplename", value ="normalized_counts",2:2)
head(top_ROSI_aggregate)

top_ROSI_aggregate<- inner_join(top_ROSI_aggregate,
                     rownames_to_column(coldata_ROSI, var ="samplename"),
                     by ="samplename")
#Annotating the genes with symbols
top_ROSI_aggregate <- left_join(x = top_ROSI_aggregate,
             y = grch38[,c("ensgene","symbol","description")],
             by ="ensgene")
ggplot(top_ROSI_aggregate)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top 50 Significant DE Genes Day")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))

####GSEA for all significant genes
ROSI_aggregate_l2fc <- res_ROSI_anno_noMSC_D10_D14$log2FoldChange
names(ROSI_aggregate_l2fc) <- res_ROSI_anno_noMSC_D10_D14$symbol
ROSI_aggregate_l2fc<-na.omit(ROSI_aggregate_l2fc)
length(ROSI_aggregate_l2fc)
ROSI_aggregate_l2fc <- ROSI_aggregate_l2fc[-which(duplicated(names(ROSI_aggregate_l2fc)))]
# sort the list in decreasing order (required for clusterProfiler)
ROSI_aggregate_l2fc = sort(ROSI_aggregate_l2fc, decreasing = TRUE)
keytypes(org.Hs.eg.db)
head(ROSI_aggregate_l2fc)

ROSI_aggregate_l2fc_up <- ROSI_aggregate_l2fc[ROSI_aggregate_l2fc>0]
length(ROSI_aggregate_l2fc_up)#639 up regulated genes
gse_ROSI_aggregate_l2fc_up <- gseGO(geneList=ROSI_aggregate_l2fc_up, 
             ont = "All", 
             keyType = "SYMBOL", 
             minGSSize = 3, 
             maxGSSize = 300, 
             pvalueCutoff = 0.05,
             nPermSimple= 10000,
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "none")
gse_ROSI_aggregate_l2fc_up
dotplot(gse_ROSI_aggregate_l2fc_up,showCategory=10) + ggtitle("GSEA ROSI vs DMSO aggregated list")
#### Reduce clusters######
library(rrvgo)
simMatrix_ROSI_aggregate_l2fc_up <- calculateSimMatrix(gse_ROSI_aggregate_l2fc_up@result$ID, orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")

scores_simMatrix_ROSI_aggregate_l2fc_up <- setNames(-log10(gse_ROSI_aggregate_l2fc_up@result$qvalues), gse_ROSI_aggregate_l2fc_up@result$ID)
reducedTerms_ROSI_aggregate_l2fc_up <- reduceSimMatrix(simMatrix_ROSI_aggregate_l2fc_up, scores_simMatrix_ROSI_aggregate_l2fc_up, threshold = 0.7, orgdb = "org.Hs.eg.db")
#heatmapPlot(simMatrix_ROSI_D14,reducedTerms_ROSI_D14,annotateParent = T,annotationLabel = "parentTerm",fontsize= 6) # this is super noisy
scatterPlot(simMatrix_ROSI_aggregate_l2fc_up,reducedTerms_ROSI_aggregate_l2fc_up)
treemapPlot(reducedTerms_ROSI_aggregate_l2fc_up)
reducedTerms_ROSI_aggregate_l2fc_up_scoreabove0 <- reducedTerms_ROSI_aggregate_l2fc_up[which(reducedTerms_ROSI_aggregate_l2fc_up$score>0),]
scatterPlot(simMatrix_ROSI_aggregate_l2fc_up,reducedTerms_ROSI_aggregate_l2fc_up_scoreabove0)
treemapPlot(reducedTerms_ROSI_aggregate_l2fc_up_scoreabove0)
###time course of sereveral genes that increase over time
#subsetting the genes that were identified by enrichR Chip-X TF program
PPARG_genes_enrichR <- c("SDPR","SHPRH","N4BP2","DTWD1","ATP1A2","AQP7","FABP4","CLEC2D")
PPARD_geens_enrichR <- c("DTWD1","C1orf220","C1orf101","MALAT1")
dim(res_ROSI_anno_noMSC_D10_D14) #1160 7 
head(res_ROSI_anno_noMSC_D10_D14)

res_sig_ROSI_PPAR_subset <- res_ROSI_anno_noMSC_D10_D14[which(res_ROSI_anno_noMSC_D10_D14$symbol %in% c(PPARG_genes_enrichR, PPARD_geens_enrichR)),]#make PPARG subset for overall/D10/D14

head(res_sig_ROSI_PPAR_subset)
view(res_ROSI_anno_noMSC_D10_D14)
dim(vsd_mat_ROSI)
res_sig_ROSI_PPAR_subset_NC<- vsd_mat_ROSI[which(rownames(vsd_mat_ROSI) %in% rownames(res_sig_ROSI_PPAR_subset)),]#normalized counts for PPARF subset overall/D10/D14
head(res_sig_ROSI_PPAR_subset_NC)
dim(res_sig_ROSI_PPAR_subset_NC)
c(res_sig_ROSI_PPAR_subset$symbol,rownames(res_sig_ROSI_PPAR_subset))
rownames(res_sig_ROSI_PPAR_subset_NC) <- res_sig_ROSI_PPAR_subset$symbol[1:11]
ROSI_pparg_TFs <- rownames(res_sig_ROSI_PPAR_subset_NC)

#Subset PPAR only overall DEGs
res_sig_ROSI_noMSC_PPAR_subset <- ROSI_DEGS_anno_padj0.05[which(ROSI_DEGS_anno_padj0.05$symbol %in% c(PPARG_genes_enrichR, PPARD_geens_enrichR)),]
dim(res_sig_ROSI_noMSC_PPAR_subset) # only 3 genes were overall DEGS
res_sig_ROSI_noMSC_PPAR_subset

#Subset PPAR D10 DEGs
res_sig_ROSI_D10_PPAR_subset <- res_ROSI_contrD10_anno_padj_0.05[which(res_ROSI_contrD10_anno_padj_0.05$symbol %in% c(PPARG_genes_enrichR, PPARD_geens_enrichR)),]
dim(res_sig_ROSI_D10_PPAR_subset) #8 7
#subset PPAR D14 DEGs 
res_sig_ROSI_D14_PPAR_subset <- res_ROSI_contrD14_anno_padj_0.05[which(res_ROSI_contrD14_anno_padj_0.05$symbol %in% c(PPARG_genes_enrichR, PPARD_geens_enrichR)),]
res_sig_ROSI_D14_PPAR_subset #2 by 7
###heatmap of PPARg PPARD genes####

pheatmap(res_sig_ROSI_PPAR_subset_NC,
         color = heat_colors,
         cluster_rows =T,
         cluster_cols = F,
         show_rownames =T,
         annotation = select(coldata_ROSI, condition,Day),
         scale ="row")
reorder <- c(which(substr(colnames(res_sig_ROSI_PPAR_subset_NC),2,2) =="D"),which(substr(colnames(res_sig_ROSI_PPAR_subset_NC),2,2) =="R"))
res_sig_ROSI_PPARG_reorderedNC <- res_sig_ROSI_PPAR_subset_NC[,reorder]
library(ComplexHeatmap)
dim(res_sig_ROSI_PPARG_reorderedNC)
res_sig_ROSI_PPARG_reorderedNC_mat <- as.matrix(res_sig_ROSI_PPARG_reorderedNC,rownames = T, colnames = T)
class(res_sig_ROSI_PPARG_reorderedNC_mat)
head(res_sig_ROSI_PPARG_reorderedNC_mat)
Heatmap(res_sig_ROSI_PPARG_reorderedNC_mat,
         color = heat_colors,
         cluster_rows =T,
         column_order= colnames(res_sig_ROSI_PPARG_reorderedNC_mat),
         show_row_names =T,
         top_annotation = select(coldata_ROSI, condition,Day),
        column_split = c(rep("DMSO",13),rep("ROSI",14))
        )
Heatmap(t(scale(t(res_sig_ROSI_PPARG_reorderedNC_mat))),color = heat_colors,cluster_rows = T,column_order= colnames(res_sig_ROSI_PPARG_reorderedNC_mat),show_row_names = T,column_split = c(rep("DMSO",13),rep("ROSI",14)))

dim(res_sig_ROSI_PPARG_reorderedNC)
###Plot genes over Time Regressed#####
# get the counts for the gen with the lowest p value over each time point
SDPR_Time <- plotCounts(DES_ROSI, gene = "ENSG00000168497", 
                   intgroup = c("Day","condition"), returnData = TRUE)
#Plot the most sig DEG over time

SDPR_smooth <- ggplot(SDPR_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_smooth() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("SDPR")
SDPR_smooth
SDPR_dotplot <- ggplot(SDPR_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_point() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("SDPR") +
  coord_cartesian(xlim = c(2,15))
SDPR_dotplot
SDPR_Time$Day <- factor(SDPR_Time$Day)
SDPR_boxplot <- ggplot(SDPR_Time,
  aes(x = Day, y = count, fill = condition)) + 
  geom_boxplot() + stat_summary(fun=median, geom="line",aes(group= condition,colour = condition)) +
  ggtitle("SDPR")
SDPR_boxplot

###FABP4####
FABP4_Time <- plotCounts(DES_ROSI, gene = "ENSG00000170323", 
                   intgroup = c("Day","condition"), returnData = TRUE)
#Plot the most sig DEG over time
FABP4_Time$Day <- as.numeric(as.character(FABP4_Time$Day))
FABP4_smooth <- ggplot(FABP4_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_smooth() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("FABP4 DMSO vs ROSI")
FABP4_smooth
FABP4_dotplot <- ggplot(FABP4_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_point() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("FABP4 DMSO vs ROSI") +
  coord_cartesian(xlim = c(2,15))
FABP4_dotplot


###ATP1A2####
ATP1A2_Time <- plotCounts(DES_ROSI, gene = "ENSG00000018625", 
                   intgroup = c("Day","condition"), returnData = TRUE)
#Plot the most sig DEG over time
ATP1A2_Time$Day <- as.numeric(as.character(ATP1A2_Time$Day))
ATP1A2_smooth <- ggplot(ATP1A2_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_smooth() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("ATP1A2 DMSO vs ROSI")
ATP1A2_smooth
ATP1A2_dotplot <- ggplot(ATP1A2_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_point() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("ATP1A2 DMSO vs ROSI") +
  coord_cartesian(xlim = c(2,15))
ATP1A2_dotplot

###CLEC2D####
CLEC2D_Time <- plotCounts(DES_ROSI, gene = "ENSG00000069493", 
                   intgroup = c("Day","condition"), returnData = TRUE)
#Plot the most sig DEG over time
CLEC2D_Time$Day <- as.numeric(as.character(CLEC2D_Time$Day))
CLEC2D_smooth <- ggplot(CLEC2D_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_smooth() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("CLEC2D DMSO vs ROSI")
CLEC2D_smooth
CLEC2D_dotplot <- ggplot(CLEC2D_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_point() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("CLEC2D DMSO vs ROSI") +
  coord_cartesian(xlim = c(2,15))
CLEC2D_dotplot

###DTWD1####
DTWD1_Time <- plotCounts(DES_ROSI, gene = "ENSG00000104047", 
                   intgroup = c("Day","condition"), returnData = TRUE)
#Plot the most sig DEG over time
DTWD1_Time$Day <- as.numeric(as.character(DTWD1_Time$Day))
DTWD1_smooth <- ggplot(DTWD1_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_smooth() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("DTWD1 DMSO vs ROSI")
DTWD1_smooth
DTWD1_dotplot <- ggplot(DTWD1_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_point() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("DTWD1 DMSO vs ROSI") +
  coord_cartesian(xlim = c(2,15))
DTWD1_dotplot

###AQP7####
AQP7_Time <- plotCounts(DES_ROSI, gene = "ENSG00000165269", 
                   intgroup = c("Day","condition"), returnData = TRUE)
#Plot the most sig DEG over time
AQP7_Time$Day <- as.numeric(as.character(AQP7_Time$Day))
AQP7_smooth <- ggplot(AQP7_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_smooth() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("AQP7 DMSO vs ROSI")
AQP7_smooth
AQP7_dotplot <- ggplot(AQP7_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_point() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("AQP7 DMSO vs ROSI") +
  coord_cartesian(xlim = c(2,15))
AQP7_dotplot

###SHPRH####
SHPRH_Time <- plotCounts(DES_ROSI, gene = "ENSG00000146414", 
                   intgroup = c("Day","condition"), returnData = TRUE)
#Plot the most sig DEG over time
SHPRH_Time$Day <- as.numeric(as.character(SHPRH_Time$Day))
SHPRH_smooth <- ggplot(SHPRH_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_smooth() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("SHPRH DMSO vs ROSI")
SHPRH_smooth
SHPRH_dotplot <- ggplot(SHPRH_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_point() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("SHPRH DMSO vs ROSI") +
  coord_cartesian(xlim = c(2,15))
SHPRH_dotplot

###N4BP2####
N4BP2_Time <- plotCounts(DES_ROSI, gene = "ENSG00000078177", 
                   intgroup = c("Day","condition"), returnData = TRUE)
#Plot the most sig DEG over time
N4BP2_Time$Day <- as.numeric(as.character(N4BP2_Time$Day))
N4BP2_smooth <- ggplot(N4BP2_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_smooth() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("N4BP2 DMSO vs ROSI")
N4BP2_smooth
N4BP2_dotplot <- ggplot(N4BP2_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_point() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("N4BP2 DMSO vs ROSI") +
  coord_cartesian(xlim = c(2,15))
N4BP2_dotplot
###PPARD genes#####
###MALAT1####
MALAT1_Time <- plotCounts(DES_ROSI, gene = "ENSG00000251562", 
                   intgroup = c("Day","condition"), returnData = TRUE)
#Plot the most sig DEG over time
MALAT1_Time$Day <- as.numeric(as.character(MALAT1_Time$Day))
MALAT1_smooth <- ggplot(MALAT1_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_smooth() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("MALAT1 DMSO vs ROSI")
MALAT1_smooth
MALAT1_dotplot <- ggplot(MALAT1_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_point() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("MALAT1 DMSO vs ROSI") +
  coord_cartesian(xlim = c(2,15))
MALAT1_dotplot

###C1orf220####
C1orf220_Time <- plotCounts(DES_ROSI, gene = "ENSG00000213057", 
                   intgroup = c("Day","condition"), returnData = TRUE)
#Plot the most sig DEG over time
C1orf220_Time$Day <- as.numeric(as.character(C1orf220_Time$Day))
C1orf220_smooth <- ggplot(C1orf220_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_smooth() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("C1orf220 DMSO vs ROSI")
C1orf220_smooth
C1orf220_dotplot <- ggplot(C1orf220_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_point() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("C1orf220 DMSO vs ROSI") +
  coord_cartesian(xlim = c(2,15))
C1orf220_dotplot

###C1orf101####
C1orf101_Time <- plotCounts(DES_ROSI, gene = "ENSG00000179397", 
                   intgroup = c("Day","condition"), returnData = TRUE)
#Plot the most sig DEG over time
C1orf101_Time$Day <- as.numeric(as.character(C1orf101_Time$Day))
C1orf101_smooth <- ggplot(C1orf101_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_smooth() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("C1orf101 DMSO vs ROSI")
C1orf101_smooth
C1orf101_dotplot <- ggplot(C1orf101_Time,
  aes(x = Day, y = count, color = condition, group = condition)) + 
  geom_point() + stat_summary(fun=mean, geom="line") +
  scale_y_log10() + ggtitle("C1orf101 DMSO vs ROSI")
C1orf101_dotplot

###Plot all on the same pdf
PPARD_geens_enrichR # "DTWD1"    "C1orf220" "C1orf101" "MALAT1"  
PPARG_genes_enrichR #"SDPR"   "SHPRH"  "N4BP2"  "DTWD1"  "ATP1A2" "AQP7"   "FABP4"  "CLEC2D"
SDPR_dotplot
library(ggplot2)
library(GGally)
####PPARG plots######
SDPR_Time$Day <- factor(SDPR_Time$Day)
SDPR_boxplot <- ggplot(SDPR_Time,
  aes(x = Day, y = count, fill = condition)) + 
  geom_boxplot() + stat_summary(fun=median, geom="line",aes(group= condition,colour = condition)) +
  ggtitle("SDPR")
SDPR_boxplot

SHPRH_Time$Day <- factor(SHPRH_Time$Day)
SHPRH_boxplot <- ggplot(SHPRH_Time,
  aes(x = Day, y = count, fill = condition)) + 
  geom_boxplot() + stat_summary(fun=median, geom="line",aes(group= condition,colour = condition)) +
  ggtitle("SHPRH")
SHPRH_boxplot
SDPR_Time$Day <- factor(SDPR_Time$Day)
SDPR_Time$gene <- "SDPR"
SHPRH_Time$Day <- factor(SHPRH_Time$Day)
SHPRH_Time$gene <- "SHPRH"
N4BP2_Time$Day <- factor(N4BP2_Time$Day)
N4BP2_Time$gene <- "N4BP2"
DTWD1_Time$Day <- factor(DTWD1_Time$Day)
DTWD1_Time$gene <- "DTWD1"
ATP1A2_Time$Day <- factor(ATP1A2_Time$Day)
ATP1A2_Time$gene <- "ATP1A2"
AQP7_Time$Day <- factor(AQP7_Time$Day)
AQP7_Time$gene <- "AQP7"
FABP4_Time$Day <- factor(FABP4_Time$Day)
FABP4_Time$gene <- "FABP4"
CLEC2D_Time$Day <- factor(CLEC2D_Time$Day)
CLEC2D_Time$gene <- "CLEC2D"
PPARG_Time <- rbind(SDPR_Time,SHPRH_Time,N4BP2_Time,DTWD1_Time,ATP1A2_Time,AQP7_Time,FABP4_Time,CLEC2D_Time)
PPARG_boxplot <- ggplot(PPARG_Time,
  aes(x = Day, y = count, fill = condition)) + 
  geom_boxplot() + 
  stat_summary(fun=median, geom="line",aes(group= condition,colour = condition)) +
  ggtitle("PPARG Genes")+
  facet_wrap(~gene)

PPARG_boxplot

PPARG_dotplot <- ggplot(PPARG_Time,
  aes(x = Day, y = count,color = condition, group = condition)) + 
  geom_point() + 
  stat_summary(fun=median, geom="line",aes(group= condition,colour = condition)) +
  ggtitle("PPARG Genes")+ geom_point() + 
  scale_y_log10() +
  facet_wrap(~gene)

PPARG_dotplot
####PPARD plots######
PPARD_geens_enrichR # "DTWD1"    "C1orf220" "C1orf101" "MALAT1" 

DTWD1_Time$Day <- factor(DTWD1_Time$Day)
DTWD1_Time$gene <- "DTWD1"
C1orf101_Time$Day <- factor(C1orf101_Time$Day)
C1orf101_Time$gene <- "C1orf101"
C1orf220_Time$Day <- factor(C1orf220_Time$Day)
C1orf220_Time$gene <- "C1orf220"
MALAT1_Time$Day <- factor(MALAT1_Time$Day)
MALAT1_Time$gene <- "MALAT1"

PPARD_Time <- rbind(C1orf101_Time,C1orf220_Time,DTWD1_Time,MALAT1_Time)
PPARD_boxplot <- ggplot(PPARD_Time,
  aes(x = Day, y = count, fill = condition)) + 
  geom_boxplot() + 
  stat_summary(fun=median, geom="line",aes(group= condition,colour = condition)) +
  ggtitle("PPARD Genes")+
  facet_wrap(~gene)

PPARD_boxplot

PPARD_dotplot <- ggplot(PPARD_Time,
  aes(x = Day, y = count,color = condition, group = condition)) + 
  geom_point() + 
  stat_summary(fun=median, geom="line",aes(group= condition,colour = condition)) +
  ggtitle("PPARD Genes")+ geom_point() + 
  scale_y_log10() +
  facet_wrap(~gene)

PPARD_dotplot

```

#### Daily analysis continued
Given that there are so few genes that are altered between the DMSO and ROSI samples at ll days except 10, I am going to try creating lists of the top up and down regulated genes from each subset, and seeing how they look, like before, then running them through enrichR in those subsets to see if directionality helps to give some context to the TF part of the data. 
```{r up vs down regulated genes by day}
####Day 3 Up vs Down####
##Significant in each tables
res_ROSI_MSC_TC_sig_down <-res_contr_ROSI__MSC_TC_padj_0.1[which(res_contr_ROSI__MSC_TC_padj_0.1$log2FoldChange<0),] 
dim(res_ROSI_MSC_TC_sig_down) #1 6
res_ROSI_MSC_TC_sig_up <-res_contr_ROSI__MSC_TC_padj_0.1[which(res_contr_ROSI__MSC_TC_padj_0.1$log2FoldChange>0),]
dim(res_ROSI_MSC_TC_sig_up) #0 6

#Normalized counts table for up and down regulations
res_sig_down_normalized_counts <- vsd_mat_ROSI_MSC_TC[rownames(res_ROSI_MSC_TC_sig_down),]
res_sig_up_normalized_counts <- vsd_mat_ROSI_MSC_TC[rownames(res_ROSI_MSC_TC_sig_up),]
    

```
```{r Saving Variables up to this point, eval= FALSE}
save(list = c("res_sig_order_ROSI_MSC_TC","res_sig_normalized_counts_ROSI_MSC_TC","res_ROSI_MSC_TC_sig_down","res_ROSI_MSC_TC_sig_up","normalized_counts_ROSI_MSC_TC","DES_ROSI_MSC_TC","res_contr_ROSI_MSC_TC","res_contr_ROSI_MSC_TC","vsd_ROSI_MSC_TC","res_ROSI_MSC_TC","dds_ROSI_MSC_TC","cts_mat_ROSI_MSC_TC","coldata_ROSI_MSC_TC"),file = "~/Documents/Git/MSC_Bulk/RNA_seq_Data/ROSI_MSC_TC_data.Rdata")
```

```{r Normalized significant genes Heatmaps}
#head(res_SD_D0_D14_sig)
# Choose a color palette from RColorBrewer
library(RColorBrewer)
heat_colors <- brewer.pal(6,"YlOrRd")
#display.brewer.all()

# Run pheatmap
pheatmap(res_sig_normalized_counts_ROSI_MSC_TC,
         color = heat_colors,
         cluster_rows =T,
         show_rownames =F,
         annotation = select(coldata_ROSI, Time),
         scale ="row")
```
interestingly we see the MSCs segregate out, then the ROSI samples all of them are next and then finally the rosiglitazone and control differentiation 
#Top genes
```{r top genes}
library(annotables)
#annotate the genes

top_ROSI_MSC_TC <- data.frame(res_sig_normalized_counts_ROSI_MSC_TC) %>% rownames_to_column(var ="ensgene")
dim(top_ROSI)
top_ROSI <- gather(top_ROSI, key ="samplename", value ="normalized_counts",2:2)
head(top_50)

top_ROSI<- inner_join(top_ROSI,
                     rownames_to_column(coldata_norm, var ="samplename"),
                     by ="samplename")



#Annotating the genes with symbols
top_ROSI <- left_join(x = top_ROSI,
             y = grch38[,c("ensgene","symbol","description")],
             by ="ensgene")



ggplot(top_ROSI)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top 50 Significant DE Genes Day")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))
```

##Transcription factor subset
```{r transcription subset}
#annotate significant ROSI genes
res_sig_ROSI_anno<- as.data.frame(res_sig_normalized_counts_ROSI) %>% rownames_to_column(var ="ensgene")
res_sig_ROSI_anno <- left_join(x = res_sig_ROSI_anno,
             y = grch38[,c("ensgene","symbol","description")],
             by ="ensgene")
dim(res_sig_ROSI_anno) #114 30
####isolate transciption subset #####
trans_ROSI <- res_sig_ROSI_anno[grep(res_sig_ROSI_anno$description,pattern = "transcription"),]
trans_ROSI
```
There are only 4 transcription factors that are altered for the ROSI samples they are HLF PARbZIP, MYC, FOS(AP-1 subunit), and TWIST2


#Adipocyte Types
```{r adipocyte types}
white_genes <- c("LEP","ASC1","SLC7A10","FABP4","LPL","MPZL2","NR1H3","LXRA","NRIP1","RIP140","RB1","RBL1","RESISTIN","SERPINA3K","TCF21","WDNM1LIKE","EBF3","HOXC8","HOXC9","PDGFRA","PPARG","DICER1","CEBPA","ANGPTL4","TIMP4","AOC3","ADH1B","PREF1")
vsd_mat_ROSI_anno <- as.data.frame(vsd_mat_ROSI) %>% rownames_to_column(var ="ensgene")
vsd_mat_ROSI_anno <- left_join(x = vsd_mat_ROSI_anno,
             y = grch38[,c("ensgene","symbol","description")],
             by ="ensgene")
WG_ROSI <-grep(vsd_mat_ROSI_anno$symbol, pattern = paste(white_genes,collapse = "|"))
WG_ROSI_anno <- vsd_mat_ROSI_anno[WG_ROSI,]
length(white_genes)

Beige_genes <- c("AQP7","ASC1","CAR4","CD137","CITED1","EAR2","NR2F6","SHOX2","SLC27A1","SP100","TBX1","TMEM26","CIDEA","COX7A","EPSTI1","FGF21","HSPB7","KCNK3","LHX8","KLF11")
BeG_ROSI <-grep(vsd_mat_ROSI_anno$symbol, pattern = paste(Beige_genes,collapse = "|"))
BeG_ROSI_anno <- vsd_mat_ROSI_anno[BeG_ROSI,]
BeG_ROSI_anno <- unique(BeG_ROSI_anno)
length(Beige_genes)
Brown_genes <- c("BMP7","EBF2","EDNRB","EVA1","MIR133B","MIR206","MYF5","PDK4","PREX1","ZIC1","UCP1","PPARA","CIDEA")
BrG_ROSI <-grep(vsd_mat_ROSI_anno$symbol, pattern = paste(Brown_genes,collapse = "|"))
BrG_ROSI_anno <- vsd_mat_ROSI_anno[BrG_ROSI,]
length(Brown_genes)
#BrG_ROSI_anno
```


```{r visualizing Adpocyte types}
#Getting count data for these genes


rownames(WG_ROSI_anno) <- WG_ROSI_anno$symbol
res_ROSI_normalized_counts_WAT_exactmatch <- WG_ROSI_anno[which(rownames(WG_ROSI_anno) %in% white_genes),]

dim(res_ROSI_normalized_counts_WAT_exactmatch) #18 30
#Heat map of Gene panels
pheatmap(res_ROSI_normalized_counts_WAT_exactmatch[2:27],
         color = heat_colors,
         cluster_rows =T,
         show_rownames =T,
         annotation = select(coldata_ROSI, condition),
         scale ="row")

#Getting count data for these genes
rownames(BeG_ROSI_anno) <- BeG_ROSI_anno$symbol
#subsetting for exact gene names
res_ROSI_normalized_counts_BeAT_exactmatch <- BeG_ROSI_anno[(rownames(BeG_ROSI_anno) %in% Beige_genes),]

#Heat map of Gene panels
dim(res_ROSI_normalized_counts_BeAT_exactmatch)
pheatmap(res_ROSI_normalized_counts_BeAT_exactmatch[2,27],
         color = heat_colors,
         cluster_rows =T,
         show_rownames =T,
         annotation = select(coldata_ROSI_Noextra, Time),
         scale ="row")

#Getting count data for these genes

rownames(BrG_ROSI_anno) <- BrG_ROSI_anno$symbol
#Subsetting for Exact gene name matches
res_ROSI_normalized_counts_BAT_exactmatch <- BrG_ROSI_anno[which(rownames(BrG_ROSI_anno) %in% Brown_genes),]
#Heat map of Gene panels
dim(res_ROSI_normalized_counts_BAT_exactmatch)
pheatmap(res_ROSI_normalized_counts_BAT_exactmatch[2,27],
         color = heat_colors,
         cluster_rows =T,
         show_rownames =T,
         annotation = select(coldata_ROSI_Noextra, Time),
         scale ="row")
```


#Analysis Time + Condition
This analysis is the same as the basic one but I had the analysis take time into consideration. 

```{r DDS object Design Time/condition, eval=FALSE}
#Trying the same differentiation but including MSCs to see if anything else pops out. 
cts_mat_ROSI_MSC<- cts_mat[,which(substr(colnames(cts_mat),2,2)=="R"|substr(colnames(cts_mat),2,2)=="D" | substr(colnames(cts_mat),1,2)=="AO")]
rownames(cts_mat_ROSI_MSC) <- substr(rownames(cts_mat_ROSI_MSC),1,15)#Removing the .# on the gene names

coldata_ROSI_MSC <- coldata_norm[which(substr(rownames(coldata_norm),2,2)=="R"|substr(rownames(coldata_norm),2,2)=="D" |substr(rownames(coldata_norm),1,2)=="AO"),]
dim(cts_mat_ROSI_MSC) #58037 30
dim(coldata_ROSI_MSC) #30 4

dds_ROSI_MSC_TC <- DESeqDataSetFromMatrix(countData = cts_mat_ROSI_MSC,
                                                 colData = coldata_ROSI_MSC,
                                                 design = ~ Time + condition
                                                 )

rownames(dds_ROSI_MSC_TC) <- substr(rownames(dds_ROSI_MSC_TC),1,15)
featureData <- data.frame(gene=substr(rownames(cts_mat_ROSI),1,15))
mcols(dds_ROSI_MSC_TC) <- DataFrame(mcols(dds_ROSI_MSC_TC), featureData)
mcols(dds_ROSI_MSC_TC)
featureData$symbol <- cts_1021$SYMBOL# Adding mcol symbol data
mcols(dds_ROSI_MSC_TC) <- DataFrame(mcols(dds_ROSI_MSC_TC), featureData)
mcols(dds_ROSI_MSC_TC)
# Pre- filtering. Pre-filtering of count data, while not necessary for
# analysis at this point, can help to minimize processing time of DESeq
# functions!

## Filtering DDS Object
keep <- rowSums(counts(dds_ROSI_MSC_TC)) >= 50 #30255 genes passed this low bar
dds_ROSI_MSC_TC <- dds_ROSI_MSC_TC[keep,]

# 
# Setting factor levels. It can be useful to designate factor levels while
# formatting your dds object since it will influence the manner in which
# DESeq compares different groups within your data.
# 
# You can specify levels either using factor()

dds_ROSI_MSC_TC$condition <- factor(dds_ROSI_MSC_TC$condition, levels = c("control","ROSI"))
dds_ROSI_MSC_TC$condition


####NORMALIZE: This is self-explanitory. This will normalize the data so#####
#
# Determine the size factors to use for normalization
dds_ROSI_MSC_TC<- estimateSizeFactors(dds_ROSI_MSC_TC)
sizeFactors(dds_ROSI_MSC_TC)
# Extract the normalized counts
normalized_counts_ROSI_MSC_TC <- counts(dds_ROSI_MSC_TC, normalized=T)
head(normalized_counts_ROSI_MSC_TC)
```

##Clustering Variace
VST stands for Variance stablizind transformation which is based on negative binomial data with a dispetion mean trend. 
Genes with high counts will haev similar VST and Rlog results to the ordinary log2 tranformaiton of the normalized counts. Genes with lower counts will be flattened out this allows us to use the VST and rlog data for better PCAs. 
###VST vs rlog
VST is much faster ot compute and less sensitive to outliers than the rlog. rlog is better for smaller datasets with samples sizeds below ~30. you can perform both VST and rlog if you are unsure and then compare the meanSdPlot  or PCAs to see which generates better data. 
```{r correlation heatmaps}
suppressPackageStartupMessages(library(pheatmap))

# Transform the normalized counts 
vsd_ROSI_MSC_TC <- vst(dds_ROSI_MSC_TC,blind=F )

# Extract the matrix of transformed counts
vsd_mat_ROSI_MSC_TC <- assay(vsd_ROSI_MSC_TC)

# Compute the correlation values between samples
vsd_cor_ROSI_MSC_TC <- cor(vsd_mat_ROSI_MSC_TC) 

# Plot the heatmap
pheatmap(vsd_cor_ROSI_MSC_TC)
```

\#PCA time

```{r pca of all ROSI MSC}
# Transform the normalized counts 
#we did this last time so we can just use it again
#vsd_ROSI <- vst(dds_ROSI,blind=T)

#determine what to dolor the data by 
colData(vsd_ROSI_MSC_TC)
# Plot the PCA of PC1 and PC2
plotPCA(vsd_ROSI_MSC_TC, intgroup= "condition")
```

\#Looking at the other PCs

```{r}
plotPCA(vsd_ROSI_MSC_TC, intgroup = c("Time","condition"))
```

\#Checing Dispersion

```{r Dispersion, eval=FALSE}
# Plot dispersions
# This is not working for me and I am not sure why 
plotDispEsts(dds_ROSI_MSC_TC)
```
```{r Differential Expression calc, echo=FALSE, eval= FALSE}
#We didn't need to remave any of the samples so the basic analyis will be the same by conditon
# dds_ROSI <- DESeqDataSetFromMatrix(countData = cts_mat,
#                                                  colData = coldata_norm,
#                                                  design = ~ condition
#                                                  )
DES_ROSI_MSC_TC <- DESeq(dds_ROSI_MSC_TC)
res_ROSI_MSC_TC <- results(DES_ROSI_MSC_TC)
head(res_ROSI_MSC_TC)
mcols(res_ROSI_MSC_TC)
```
## Filtering results

I am adding a specific contrast and alpha values to help remove some of the extra low hanging fruit.  
```{r Differential Expression results}
res_contr_ROSI_MSC_TC <- results(DES_ROSI_MSC_TC, contrast=c("condition","ROSI","control"),alpha = 0.05)
head(res_contr_ROSI_MSC_TC)
```

Log fold change shrinkage for visualization and ranking This will help to show which genes are important and which aren't.

```{r Differential Expression logfold, eval=FALSE}
#BiocManager::install("apeglm")
#library(apeglm)
# Shrink the log2 fold change estimates to be more accurate
res_contr_ROSI_MSC_TC_lfc <- results(DES_ROSI_MSC_TC, 
                        contrast=c("condition","ROSI","control"),
                        alpha = 0.05,
                        lfcThreshold = 0.2)
resultsNames(res_contr_ROSI_MSC_TC_lfc)

resLFC <- lfcShrink(dds,
                    contrast = c("condition","ROSI","control"),
                    res = res_cont_ROSI_lfc)
summary(res_cont_ROSI_lfc)
#Save results as a dataframe
res_cont_ROSI_alldays_lfc.2 <- data.frame(res_cont_ROSI_lfc)
```


p-values and adjusted p-values

Reorganize your results by p-value, place the smaller p-values at the
top of your results list.

```{r Order results by padjusted values}
#res_Ordered <- res_cont_ROSI[order(res_cont_ROSI$pvalue),]
#summary(res_Ordered)
```

This is a lot of differentially expressed genes I will run through these
really quickly and then rerun them with a more stringent alpha and an
log2fold change cutoff added.

```{r, echo=FALSE}
sum(res_contr_ROSI_MSC_TC$padj < 0.05, na.rm=TRUE) #147 lowering the Padjusted value removed some of the genes
sum(res_contr_ROSI_MSC_TC$padj < 0.1, na.rm=TRUE) #289
res_contr_ROSI__MSC_TC_padj_0.1 <- res_contr_ROSI_MSC_TC[which(res_contr_ROSI_MSC_TC$padj<0.1),]
sum(is.na(res_contr_ROSI__MSC_TC_padj_0.1$padj))
```

##Annotate Results

```{r Annotate, eval=FALSE}
#install.packages("devtools")
#devtools::install_github("stephenturner/annotables")
library(dbplyr)
library(annotables)
grch38 # This is the genome build that I will be using for the annotation

res_cont_ROSI_MSC_anno <- data.frame(res_contr_ROSI_MSC_TC) %>% rownames_to_column(var = "genes")
res_cont_ROSI_MSC_anno$ensgene <- substr(res_cont_ROSI_MSC_anno$genes,1,15)
 
res_cont_ROSI_MSC_anno <- left_join(x=res_cont_ROSI_MSC_anno, y = grch38[,c("ensgene","symbol","description")],  by="ensgene")

head(res_cont_ROSI_MSC_anno)
ROSI_only_sig <- res_cont_ROSI_all[res_cont_ROSI_all$padj<0.5,]
ROSI_only_sig
```
```{r look at leptin}
head(res_cont_ROSI_MSC_anno)
lepsrescont <- grep(res_cont_ROSI_MSC_anno$symbol, pattern = "LEP")
res_cont_ROSI_MSC_anno[lepsrescont,]

```

####MA and volcano plots

```{r, echo=FALSE}
plotMA(res_contr_ROSI__MSC_TC_padj_0.1, ylim=c(-4,4))
head(order(res_contr_ROSI_padj_0.1,padj))
head(res_contr_ROSI_padj_0.1)

```

Example of most significantly altered genes
```{r, echo=FALSE}
plotCounts(DES_ROSI_MSC_TC, gene=which.min(res_contr_ROSI__MSC_TC_padj_0.1$padj), intgroup="condition")
```

```{r Data transformations and visualization}
vsd_ROSI_MSC_ordered <- vst(DES_ROSI_MSC_TC, blind=FALSE)
rld_ROSI_MSC_TC <- rlog(DES_ROSI_MSC_TC, blind=FALSE)
head(assay(vsd_ROSI_MSC_ordered), 5)
```
```{r variance}
#BiocManager::install("vsn")
# this gives log2(n + 1)
ntd_ROSI_MSC_TC <- normTransform(DES_ROSI_MSC_TC)
suppressPackageStartupMessages(library("vsn"))
  meanSdPlot(assay(ntd_ROSI_MSC_TC))
```

```{r Heatmap of the count matrix}
suppressPackageStartupMessages(library("pheatmap"))
select <- order(rowMeans(counts(DES_ROSI_MSC_TC,normalized=TRUE)),
                decreasing=TRUE)[1:20]

df <- as.data.frame(colData(DES_ROSI_MSC_TC)[,c("condition","Time")])
pheatmap(assay(ntd_ROSI_MSC_TC)[select,], cluster_rows=TRUE, show_rownames=FALSE,
         cluster_cols=TRUE, annotation_col=df)
```
This is to see how similar the samples are
#Visualization Results 
The above work is all qualty control for the comparison of all of hte data. I will start looking at the differentially called genes for this comparison with  creating heatmaps of differentially expressed genes and then a chart of how they are being altered. 
##Heatmap  

```{r Significant normalized counts,eval= F}
#All significant normalized counts table
res_sig_order_ROSI_MSC_TC <-res_contr_ROSI__MSC_TC_padj_0.1[order(res_contr_ROSI__MSC_TC_padj_0.1$padj),]
res_sig_normalized_counts_ROSI_MSC_TC <- vsd_mat_ROSI_MSC_TC[substr(rownames(res_contr_ROSI__MSC_TC_padj_0.1),1,15),] 

##Significant in each tables
res_ROSI_MSC_TC_sig_down <-res_contr_ROSI__MSC_TC_padj_0.1[which(res_contr_ROSI__MSC_TC_padj_0.1$log2FoldChange<0),] 
dim(res_ROSI_MSC_TC_sig_down) #1 6
res_ROSI_MSC_TC_sig_up <-res_contr_ROSI__MSC_TC_padj_0.1[which(res_contr_ROSI__MSC_TC_padj_0.1$log2FoldChange>0),]
dim(res_ROSI_MSC_TC_sig_up) #0 6

#Normalized counts table for up and down regulations
res_sig_down_normalized_counts <- vsd_mat_ROSI_MSC_TC[rownames(res_ROSI_MSC_TC_sig_down),]
res_sig_up_normalized_counts <- vsd_mat_ROSI_MSC_TC[rownames(res_ROSI_MSC_TC_sig_up),]
    

```
```{r Saving Variables up to this point, eval= FALSE}
save(list = c("res_sig_order_ROSI_MSC_TC","res_sig_normalized_counts_ROSI_MSC_TC","res_ROSI_MSC_TC_sig_down","res_ROSI_MSC_TC_sig_up","normalized_counts_ROSI_MSC_TC","DES_ROSI_MSC_TC","res_contr_ROSI_MSC_TC","res_contr_ROSI_MSC_TC","vsd_ROSI_MSC_TC","res_ROSI_MSC_TC","dds_ROSI_MSC_TC","cts_mat_ROSI_MSC_TC","coldata_ROSI_MSC_TC"),file = "~/Documents/Git/MSC_Bulk/RNA_seq_Data/ROSI_MSC_TC_data.Rdata")
```

```{r Normalized significant genes Heatmaps}
#head(res_SD_D0_D14_sig)
# Choose a color palette from RColorBrewer
library(RColorBrewer)
heat_colors <- brewer.pal(6,"YlOrRd")
#display.brewer.all()

# Run pheatmap
pheatmap(res_sig_normalized_counts_ROSI_MSC_TC,
         color = heat_colors,
         cluster_rows =T,
         show_rownames =F,
         annotation = select(coldata_ROSI, Time),
         scale ="row")
```
interestingly we see the MSCs segregate out, then the ROSI samples all of them are next and then finally the rosiglitazone and control differentiation 
#Top genes
```{r top genes}
library(annotables)
#annotate the genes

top_ROSI_MSC_TC <- data.frame(res_sig_normalized_counts_ROSI_MSC_TC) %>% rownames_to_column(var ="ensgene")
dim(top_ROSI)
top_ROSI <- gather(top_ROSI, key ="samplename", value ="normalized_counts",2:2)
head(top_50)

top_ROSI<- inner_join(top_ROSI,
                     rownames_to_column(coldata_norm, var ="samplename"),
                     by ="samplename")



#Annotating the genes with symbols
top_ROSI <- left_join(x = top_ROSI,
             y = grch38[,c("ensgene","symbol","description")],
             by ="ensgene")



ggplot(top_ROSI)+
  geom_point(aes(x = symbol, y = normalized_counts, color = condition))+
  scale_y_log10()+
  xlab("Genes")+
  ylab("Normalized Counts")+
  ggtitle("Top 50 Significant DE Genes Day")+
  theme_bw()+
  theme(axis.text.x = element_text(angle =45, hjust =1))+
  theme(plot.title = element_text(hjust =0.5))
```

#Checking for White/Beige/Brown Adipose markers
One thing that I wanted to know is if my samples are closer to one lineage of adipocytes. To do this I will look at expression of crucial adipogenic markers and segment them by adipocyte type. This paper does a comprehensive look at several [markers seen in each type of gene][https://www.frontiersin.org/articles/10.3389/fendo.2021.599134/full], this [venndiagram][https://www.frontiersin.org/files/Articles/599134/fendo-12-599134-HTML/image_m/fendo-12-599134-g001.jpg] shows where the gnes fall
 * White:
    * OB(leptin)
    * Asc1/SLC7a10
    * Fabp4
    * Lpl
    * Mpzl2
    * Nr1H3/Lxra
    * Nrip1/Rip140
    * Rb1
    * Rbl1
    * Resistin
    * Serpina3K
    Tcf21
    Wdnm1Like
  white/Beige:
    Ebf3
    Hoxc8
    Hoxc9
    Pdgfralpha
  beige
```{r gene expression of adipocyte types}
head(res_SD_NoExtra_alph05_anno)
vsd_mat_ROSI_MSC
vsd_mat_ROSI_MSC_anno <- as.data.frame(vsd_mat_ROSI_MSC) %>% rownames_to_column(var ="ensgene")
vsd_mat_ROSI_MSC_anno <- left_join(x = vsd_mat_ROSI_MSC_anno,
             y = grch38[,c("ensgene","symbol","description")],
             by ="ensgene")
WG_ROSI_MSC <-grep(vsd_mat_ROSI_MSC_anno$symbol, pattern = paste(white_genes,collapse = "|"))
WG_ROSI_MSC_anno <- vsd_mat_ROSI_MSC_anno[WG_ROSI_MSC,]
length(white_genes)

Beige_genes <- c("AQP7","ASC1","CAR4","CD137","CITED1","EAR2","NR2F6","SHOX2","SLC27A1","SP100","TBX1","TMEM26","CIDEA","COX7A","EPSTI1","FGF21","HSPB7","KCNK3","LHX8","KLF11")
BeG_ROSI_MSC <-grep(vsd_mat_ROSI_MSC_anno$symbol, pattern = paste(Beige_genes,collapse = "|"))
BeG_ROSI_MSC_anno <- vsd_mat_ROSI_MSC_anno[BeG_ROSI_MSC,]
BeG_ROSI_MSC_anno <- unique(BeG_ROSI_MSC_anno)
length(Beige_genes)
Brown_genes <- c("BMP7","EBF2","EDNRB","EVA1","MIR133B","MIR206","MYF5","PDK4","PREX1","ZIC1","UCP1","PPARA","CIDEA")
BrG_ROSI_MSC <-grep(vsd_mat_ROSI_MSC_anno$symbol, pattern = paste(Brown_genes,collapse = "|"))
BrG_ROSI_MSC_anno <- vsd_mat_ROSI_MSC_anno[BrG_ROSI_MSC,]
length(Brown_genes)
#BrG_ROSI_MSC_anno
```


```{r visualizing Adpocyte types}
#Getting count data for these genes


rownames(WG_ROSI_MSC_anno) <- WG_ROSI_MSC_anno$symbol
res_ROSI_MSC_normalized_counts_WAT_exactmatch <- WG_ROSI_MSC_anno[which(rownames(WG_ROSI_MSC_anno) %in% white_genes),]

dim(res_ROSI_MSC_normalized_counts_WAT_exactmatch) #18 30
#Heat map of Gene panels
pheatmap(res_ROSI_MSC_normalized_counts_WAT_exactmatch[2:27],
         color = heat_colors,
         cluster_rows =T,
         show_rownames =T,
         annotation = select(coldata_ROSI_MSC, condition),
         scale ="row")

#Getting count data for these genes
rownames(BeG_ROSI_MSC_anno) <- BeG_ROSI_MSC_anno$symbol
#subsetting for exact gene names
res_ROSI_MSC_normalized_counts_BeAT_exactmatch <- BeG_ROSI_MSC_anno[(rownames(BeG_ROSI_MSC_anno) %in% Beige_genes),]

#Heat map of Gene panels
dim(res_ROSI_MSC_normalized_counts_BeAT_exactmatch)
pheatmap(res_ROSI_MSC_normalized_counts_BeAT_exactmatch[2,27],
         color = heat_colors,
         cluster_rows =T,
         show_rownames =T,
         annotation = select(coldata_ROSI_MSC_Noextra, Time),
         scale ="row")

#Getting count data for these genes

rownames(BrG_ROSI_MSC_anno) <- BrG_ROSI_MSC_anno$symbol
#Subsetting for Exact gene name matches
res_ROSI_MSC_normalized_counts_BAT_exactmatch <- BrG_ROSI_MSC_anno[which(rownames(BrG_ROSI_MSC_anno) %in% Brown_genes),]
#Heat map of Gene panels
dim(res_ROSI_MSC_normalized_counts_BAT_exactmatch)
pheatmap(res_ROSI_MSC_normalized_counts_BAT_exactmatch[2,27],
         color = heat_colors,
         cluster_rows =T,
         show_rownames =T,
         annotation = select(coldata_ROSI_MSC_Noextra, Time),
         scale ="row")

```
```{r visualizing Adpocyte typesday 0 to 14}
#Getting count data for these genes

res_SD_NE_normalized_counts_WAT <- vsd_SD_NE_D014_mat_normdata_annoDbi[rownames(WG_SD_anno),] 
rownames(res_SD_NE_normalized_counts_WAT) <- WG_SD_anno$symbol
res_SD_NE_normalized_counts_WAT_exactmatch <- vsd_SD_NE_D014_mat_normdata_annoDbi[which(vsd_SD_NE_D014_mat_normdata_annoDbi$symbol  %in% white_genes),]
rownames(res_SD_NE_normalized_counts_WAT_exactmatch) <- res_SD_NE_normalized_counts_WAT_exactmatch$symbol
#Heat map of Gene panels
pheatmap(res_SD_NE_normalized_counts_WAT_exactmatch[,c(1:3,14:16)],
         color = heat_colors,
         cluster_rows =T,
         show_rownames =T,
         annotation = select(coldata_SD_Noextra, Day),
         scale ="row")

#Getting count data for these genes

res_SD_NE_normalized_counts_BeAT <- vsd_SD_NE_D014_mat_normdata_annoDbi[rownames(BeG_SD_anno),] 
rownames(res_SD_NE_normalized_counts_BeAT) <- BeG_SD_anno$symbol
#subsetting for exact gene names
res_SD_NE_normalized_counts_BeAT_exactmatch <- vsd_SD_NE_D014_mat_normdata_annoDbi[which(vsd_SD_NE_D014_mat_normdata_annoDbi$symbol %in% Beige_genes),]
#Heat map of Gene panels
dim(res_SD_NE_normalized_counts_BeAT) #33 17
rownames(res_SD_NE_normalized_counts_BeAT_exactmatch) <- res_SD_NE_normalized_counts_BeAT_exactmatch$symbol
pheatmap(res_SD_NE_normalized_counts_BeAT_exactmatch[,c(1:3,14:16)],
         color = heat_colors,
         cluster_rows =T,
         cluter_colums= F,
         show_rownames =T,
         annotation = select(coldata_SD_Noextra, Day),
         scale ="row")

#Getting count data for these genes

res_SD_NE_normalized_counts_BAT <- normalized_counts_SD_NoExtra[rownames(BrG_SD_anno),] 
rownames(res_SD_NE_normalized_counts_BAT) <- BrG_SD_anno$symbol
#Subsetting for Exact gene name matches
res_SD_NE_normalized_counts_BAT_exactmatch <- vsd_SD_NE_D014_mat_normdata_annoDbi[which(vsd_SD_NE_D014_mat_normdata_annoDbi$symbol %in% Brown_genes),]
rownames(res_SD_NE_normalized_counts_BAT_exactmatch) <- res_SD_NE_normalized_counts_BAT_exactmatch$symbol
#Heat map of Gene panels

pheatmap(res_SD_NE_normalized_counts_BAT_exactmatch[c(1:3,14:16)],
         color = heat_colors,
         cluster_rows =T,
         show_rownames =T,
         annotation = select(coldata_SD_Noextra, Time),
         scale ="row")
```

### Day 14 only what type of Adipocytes do they look like
```{r day 14 adipocyte substye check}
res_ROSI_MSC_normalized_counts_AT <- rbind(res_ROSI_MSC_normalized_counts_WAT_exactmatch,res_ROSI_MSC_normalized_counts_BeAT_exactmatch,res_ROSI_MSC_normalized_counts_BAT_exactmatch)

dim(res_ROSI_MSC_normalized_counts_WAT_exactmatch) #18 17
dim(res_ROSI_MSC_normalized_counts_BeAT_exactmatch)#12 17
dim(res_ROSI_MSC_normalized_counts_BAT_exactmatch)#6 17
annotdf_rows <- data.frame(row.names = c(res_ROSI_MSC_normalized_counts_AT$symbol), Adipocyte_type = c(rep("White", 18), rep("Beige", 12), rep("Brown", 6)))

res_ROSI_MSC_normalized_counts_AT_d0d14 <- res_ROSI_MSC_normalized_counts_AT[,which(substr(colnames(res_ROSI_MSC_normalized_counts_AT),1,1)=="E" |substr(colnames(res_ROSI_MSC_normalized_counts_AT),1,1)=="A")]
rownames(res_ROSI_MSC_normalized_counts_AT_d0d14) <- res_ROSI_MSC_normalized_counts_AT$symbol

pheatmap(res_ROSI_MSC_normalized_counts_AT_d0d14,
         color = heat_colors,
         cluster_rows =F,
         show_rownames =T,
         annotation = select(coldata_SD_Noextra[c(1:3,14:16),], Time),
         annotation_row = annotdf_rows,
         scale ="row")
```

### diuvergent bar pot for adipogenic differentiaiton
```{r Divergen Bar plot Adipocyte type}
#Normalizing the data
res_ROSI_MSC_normalized_counts_AT_d14 <- res_ROSI_MSC_normalized_counts_AT[,which(substr(colnames(res_ROSI_MSC_normalized_counts_AT),1,1)=="E")]
SD_D14_avg_adipogenes_ROSI_MSC <- data.frame(SD_D14=rowMeans(res_ROSI_MSC_normalized_counts_AT_d14[,1:3]))
SD_D14_avg_adipogenes_ROSI_MSC <- data.frame(log2(SD_D14_avg_adipogenes_ROSI_MSC))
res_ROSI_MSC_normalized_counts_AT_d14_normed <- log2(res_ROSI_MSC_normalized_counts_AT_d14[,1:6])-SD_D14_avg_adipogenes_ROSI_MSC$SD_D14
res_ROSI_MSC_normalized_counts_AT_d14_normed

#####Divergent bar graph ############
library(matrixStats)
#Making data frame of means 
df_AT_ROSI_MSC_log2_normed <- data.frame(Day_14_DMSO=rowMeans(res_ROSI_MSC_normalized_counts_AT_d14_normed[,1:3]),Day_14_ROSI=rowMeans(res_ROSI_MSC_normalized_counts_AT_d14_normed[,4:6]),row.names = rownames(res_ROSI_MSC_normalized_counts_AT_d14_normed),gene = rownames(res_ROSI_MSC_normalized_counts_AT_d14_normed))
head(df_AT_ROSI_MSC_log2_normed)

# lock in factor level order
df_AT_ROSI_MSC_log2_normed$gene <- factor(df_AT_ROSI_MSC_log2_normed$gene, levels = df_AT_ROSI_MSC_log2_normed$gene)
#inverting the order because the data keeps plotting upside down
df_AT_ROSI_MSC_log2_normed_reorder=df_AT_ROSI_MSC_log2_normed[order(nrow(df_AT_ROSI_MSC_log2_normed):1),]
df_AT_ROSI_MSC_log2_normed_reorder
df_AT_ROSI_MSC_log2_normed_reorder$gene <- factor(df_AT_SD_NE_log2_normed_reorder$gene, levels = df_AT_ROSI_MSC_log2_normed_reorder$gene)
library(ggplot2)
###Colors
library(MetBrewer)
degascolor <- met.brewer(name = 'Morgenstern',7,type = "discrete")
###Control Adipocytes
color <- ifelse(df_AT_ROSI_MSC_log2_normed_reorder$Day_14_ROSI < 0, degascolor[6], degascolor[4])
ggplot(df_AT_ROSI_MSC_log2_normed_reorder, aes(y = gene, x = Day_14_ROSI)) +
  geom_bar(stat = "identity",
           show.legend = FALSE, fill = color) + # Remove the legend +
  xlab("ROSI log2 Relative Gene expression to D14 Standard adipocyte") +
  ylab("Day_14_DMSO") + scale_x_continuous(limits = c(-1,1))
```
##Add MSCs back to the data set
```{r ROSI analysis + MSCs}
suppressPackageStartupMessages(library("DESeq2"))

#Subsetting the data to only have the pertinent information
cts_mat_ROSI_MSC<- cts_mat[,which(substr(colnames(cts_mat),2,2)=="R"|substr(colnames(cts_mat),2,2)=="D" |substr(colnames(cts_mat),1,2)=="AO")]
rownames(cts_mat_ROSI_MSC) <- substr(rownames(cts_mat_ROSI_MSC),1,15)#Removing the .# on the gene names

coldata_ROSI_MSC <- coldata_norm[which(substr(rownames(coldata_norm),2,2)=="R"|substr(rownames(coldata_norm),2,2)=="D"|substr(rownames(coldata_norm),1,2)=="AO"),]
dim(cts_mat_ROSI_MSC) #58037 30
dim(coldata_ROSI_MSC) #30 4

##Creating DDS object from the subset data                                  

dds_ROSI_MSC <- DESeqDataSetFromMatrix(countData = cts_mat_ROSI_MSC,
                                                 colData = coldata_ROSI_MSC,
                                                 design = ~ Day + condition
                                                 )
summary(dds_ROSI_MSC)
head(dds_ROSI_MSC)   
#####Adding features and metadata columns
rownames(dds_ROSI_MSC) <- substr(rownames(dds_ROSI_MSC),1,15)
featureData <- data.frame(ensgene= substr(rownames(cts_mat_ROSI_MSC),1,15))
dim(featureData)
featureData$symbol <- cts_1021$SYMBOL# Adding mcol symbol data
mcols(dds_ROSI_MSC) <- DataFrame(mcols(dds_ROSI_MSC), featureData)
mcols(dds_ROSI_MSC)

## Filtering DDS Object
#######removing low count rows #########
keep <- rowSums(counts(dds_ROSI_MSC)) >= 50 #25240 genes passed this low bar
dds_ROSI_MSC <- dds_ROSI_MSC[keep,]
dim(dds_ROSI_MSC)

#######Factor creation ##############
dds_ROSI_MSC$condition <- factor(dds_ROSI_MSC$condition, levels = c("control","ROSI"))
dds_ROSI_MSC$condition
```

\#NORMALIZE: This is self-explanitory. This will normalize the data so
that we aren't getting a lot of outliers.

```{r normalize }
# Determine the size factors to use for normalization
dds_ROSI_MSC<- estimateSizeFactors(dds_ROSI_MSC)
sizeFactors(dds_ROSI_MSC)
# Extract the normalized counts
normalized_counts_ROSI_MSC <- counts(dds_ROSI_MSC, normalized=T)
head(normalized_counts_ROSI_MSC)
```

##Clustering Variace
VST stands for Variance stablizind trasformatio which is based o egative biomial data with a dispetio mea tred. 
Genes with high counts will haev similar VST and Rlog results to the ordinary log2 tranformaiton of the normalized counts. Genes with lower counts will be flattened out this allows us to use the VST and rlog data for better PCAs. 
###VST vs rlog
VST is much faster ot compute and less sensitive to outliers than the rlog. rlog is better for smaller datasets with samples sizeds below ~30. you can perform both VST and rlog if you are unsure and then compare the meanSdPlot  or PCAs to see which generates better data. 
```{r correlation heatmaps}
suppressPackageStartupMessages(library(pheatmap))

# Transform the normalized counts 
vsd_ROSI_MSC <- vst(dds_ROSI_MSC,blind=F )

# Extract the matrix of transformed counts
vsd_mat_ROSI_MSC <- assay(vsd_ROSI_MSC)

# Compute the correlation values between samples
vsd_cor_ROSI_MSC <- cor(vsd_mat_ROSI_MSC) 

# Plot the heatmap
pheatmap(vsd_cor_ROSI_MSC)
# Transform the normalized counts 
#we did this last time so we can just use it again
#vsd_ROSI <- vst(dds_ROSI,blind=T)

#determine what to dolor the data by 
colData(vsd_ROSI_MSC)
# Plot the PCA of PC1 and PC2
plotPCA(vsd_ROSI_MSC, intgroup= "condition")
plotPCA(vsd_ROSI_MSC, intgroup= c("Day","condition"))
```

